diff -rupN gfnrename-0.6-orig/src/main.c gfnrename-0.6/src/main.c
--- gfnrename-0.6-orig/src/main.c	2024-08-12 21:30:57.133550578 +0300
+++ gfnrename-0.6/src/main.c	2024-08-12 21:31:06.033547074 +0300
@@ -575,8 +575,8 @@ void AppUpdatePathbox ()
 	GtkTreeModel *model;
 	char txt[MAXPATH] = "";
 	int i, len, cnt;
-	combo = (GtkComboBox *) lookup_widget(GTK_WIDGET(App1), "combo_pathbox");
-	model = gtk_combo_box_get_model(combo);
+	combo = (GtkComboBoxText *) lookup_widget(GTK_WIDGET(App1), "combo_pathbox");
+	model = gtk_combo_box_get_model(GTK_COMBO_BOX(combo));
 	Gbl->InSetup = 1;
 	cnt = gtk_tree_model_iter_n_children(model, NULL);
 	for(i = cnt; i >= 0; i--)  gtk_combo_box_text_remove(combo, i);
@@ -590,7 +590,7 @@ void AppUpdatePathbox ()
 			else  txt[i] = 0;
 		}
 	}
-	gtk_combo_box_set_active(combo, 0);
+	gtk_combo_box_set_active(GTK_COMBO_BOX(combo), 0);
 	Gbl->InSetup = 0;
 }
 	
