diff -rupN osmo-0.4.4-orig/src/options_prefs.c osmo-0.4.4/src/options_prefs.c
--- osmo-0.4.4-orig/src/options_prefs.c	2022-05-23 22:05:17.851917650 +0800
+++ osmo-0.4.4/src/options_prefs.c	2022-05-23 22:06:26.487997273 +0800
@@ -128,7 +128,7 @@ gint i;
     config.toolbar_exit_button = FALSE;
     config.date_format = DATE_YYYY_MM_DD;
     config.time_format = TIME_24;
-    config.enable_systray = TRUE;
+    config.enable_systray = FALSE;
     config.start_minimised_in_systray = FALSE;
     config.ignore_day_note_events = FALSE;
     config.run_counter = 0;
@@ -145,8 +145,8 @@ gint i;
     g_strlcpy (config.spell_lang, g_getenv("LANG"), MAXNAME);
     /* FIXME: DO NOT USE xdg-open AS IT CANNOT OPEN MULTIPLE LINKS AT ONCE */
 	/*g_strlcpy (config.web_browser, "xdg-open %s", MAXHELPERCMD);*/
-	g_strlcpy (config.web_browser, "firefox %s", MAXHELPERCMD);
-	g_strlcpy (config.email_client, "xdg-email %s", MAXHELPERCMD);
+	g_strlcpy (config.web_browser, "defaultbrowser %s", MAXHELPERCMD);
+	g_strlcpy (config.email_client, "defaultemail %s", MAXHELPERCMD);
 	/* play command requires SoX */
 	/*g_strlcpy (config.sound_player, "play %s", MAXHELPERCMD);*/
 	g_strlcpy (config.sound_player, "aplay %s", MAXHELPERCMD);
@@ -160,7 +160,7 @@ gint i;
     config.cb_window_size_y = 680;
     config.ib_window_size_x = 550;
     config.ib_window_size_y = 650;
-    config.display_options = GUI_CALENDAR_SHOW_DAY_NAMES | GUI_CALENDAR_NO_MONTH_CHANGE | GUI_CALENDAR_WEEK_START_MONDAY;
+    config.display_options = GUI_CALENDAR_SHOW_DAY_NAMES | GUI_CALENDAR_NO_MONTH_CHANGE;
     config.day_notes_visible = FALSE;
     config.timeline_start = 8*60;
     config.timeline_end = 15*60;
