diff -rupN osmo-0.4.4-orig/configure osmo-0.4.4/configure
--- osmo-0.4.4-orig/configure	2022-05-24 16:01:18.312892796 +0800
+++ osmo-0.4.4/configure	2022-05-24 16:01:48.557044142 +0800
@@ -5986,7 +5986,7 @@ else
 	LIBICAL_LIBS=$pkg_cv_LIBICAL_LIBS
         { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
-	have_libical=true
+	have_libical=false
 fi
 if test "x$have_libical" = "xtrue"; then
 
@@ -6164,7 +6164,7 @@ else
 	LIBNOTIFY_LIBS=$pkg_cv_LIBNOTIFY_LIBS
         { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
-	have_libnotify7=true
+	have_libnotify7=false
 fi
 if test "x$have_libnotify7" = "xtrue"; then
 
@@ -6244,7 +6244,7 @@ else
 	LIBNOTIFY_LIBS=$pkg_cv_LIBNOTIFY_LIBS
         { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
-	have_libnotify=true
+	have_libnotify=false
 fi
     if test "x$have_libnotify" = "xtrue"; then
 
@@ -6348,7 +6348,7 @@ else
 	LIBWEBKIT_LIBS=$pkg_cv_LIBWEBKIT_LIBS
         { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
-	have_libwebkit=true
+	have_libwebkit=false
 fi
 if test "x$have_libwebkit" = "xtrue"; then
     libwebkit="yes"
