diff -Naur util-linux-2.22.2/config/config.sub util-linux-2.22.2-patched/config/config.sub
--- util-linux-2.22.2/config/config.sub	2012-09-18 02:53:51.000000000 +0000
+++ util-linux-2.22.2-patched/config/config.sub	2016-07-23 20:52:45.685261722 +0000
@@ -253,7 +253,7 @@
 	| alpha | alphaev[4-8] | alphaev56 | alphaev6[78] | alphapca5[67] \
 	| alpha64 | alpha64ev[4-8] | alpha64ev56 | alpha64ev6[78] | alpha64pca5[67] \
 	| am33_2.0 \
-	| arc | arm | arm[bl]e | arme[lb] | armv[2345] | armv[345][lb] | avr | avr32 \
+	| arc | arm | arm[bl]e | arme[lb] | armv[2345678] | armv[345678][lb] | avr | avr32 \
         | be32 | be64 \
 	| bfin \
 	| c4x | clipper \
diff -Naur util-linux-2.22.2/lib/tt.c util-linux-2.22.2-patched/lib/tt.c
--- util-linux-2.22.2/lib/tt.c	2012-12-12 20:04:47.902355093 +0000
+++ util-linux-2.22.2-patched/lib/tt.c	2016-07-23 20:51:58.358596857 +0000
@@ -22,7 +22,7 @@
 #include "widechar.h"
 #include "tt.h"
 #include "mbsalign.h"
-#include "ttyutils.h"
+//#include "ttyutils.h"
 
 struct tt_symbols {
 	const char *branch;
diff -Naur util-linux-2.22.2/term-utils/setterm.c util-linux-2.22.2-patched/term-utils/setterm.c
--- util-linux-2.22.2/term-utils/setterm.c	2012-11-30 08:01:42.950919423 +0000
+++ util-linux-2.22.2-patched/term-utils/setterm.c	2016-07-23 20:51:58.358596857 +0000
@@ -123,15 +123,6 @@
 #include "nls.h"
 #include "closestream.h"
 
-#if __GNU_LIBRARY__ < 5
-#ifndef __alpha__
-# include <linux/unistd.h>
-#define __NR_klogctl __NR_syslog
-_syscall3(int, klogctl, int, type, char*, buf, int, len);
-#else /* __alpha__ */
-#define klogctl syslog
-#endif
-#endif
 extern int klogctl(int type, char *buf, int len);
 
 /* Constants. */
diff -Naur util-linux-2.22.2/term-utils/ttymsg.c util-linux-2.22.2-patched/term-utils/ttymsg.c
--- util-linux-2.22.2/term-utils/ttymsg.c	2012-11-29 15:15:00.347642630 +0000
+++ util-linux-2.22.2-patched/term-utils/ttymsg.c	2016-07-23 20:51:58.358596857 +0000
@@ -42,6 +42,7 @@
 
 #include <sys/types.h>
 #include <sys/uio.h>
+#include <sys/param.h>
 #include <signal.h>
 #include <fcntl.h>
 #include <dirent.h>
