diff -rup Xdialog-orig/src/interface.c Xdialog/src/interface.c
--- Xdialog-orig/src/interface.c	2021-02-05 12:01:08.456894666 +0200
+++ Xdialog/src/interface.c	2021-02-05 12:02:41.852482504 +0200
@@ -2171,13 +2171,7 @@ void create_calendar(gchar *optarg, gint
 	set_backtitle(TRUE);
 	set_label(optarg, FALSE);
 
-	flags = GTK_CALENDAR_SHOW_HEADING | GTK_CALENDAR_SHOW_DAY_NAMES;
-	/* There is a bug in GTK+ v1.2.7 preventing the week numbers to show
-	 * properly (all numbers are 0 !)...
-	 */
-	if (!(gtk_major_version == 1 && gtk_minor_version == 2 &&
-	    gtk_micro_version == 7))
-		flags = flags | GTK_CALENDAR_SHOW_WEEK_NUMBERS;
+	flags = GTK_CALENDAR_SHOW_HEADING | GTK_CALENDAR_SHOW_DAY_NAMES | GTK_CALENDAR_SHOW_WEEK_NUMBERS;
 
 	Xdialog.widget1 = gtk_calendar_new();
 	gtk_box_pack_start(Xdialog.vbox, Xdialog.widget1, TRUE, TRUE, 5);
diff -rup Xdialog-orig/src/main.c Xdialog/src/main.c
--- Xdialog-orig/src/main.c	2021-02-05 12:01:08.456894666 +0200
+++ Xdialog/src/main.c	2021-02-05 12:01:28.996801339 +0200
@@ -705,10 +705,6 @@ int main(int argc, char *argv[])
 	strcpy(Xdialog.title, XDIALOG);			/* Default widget title */
 	strcpy(Xdialog.separator, "/");			/* Default results separator */
 
-	if (gtk_major_version < 1 ||
-	    (gtk_major_version == 1 && gtk_minor_version < 2))
-		fprintf(stderr,
-			"%s: GTK+ version too old, please upgrade !\n", argv[0]);
 	gtk_set_locale();
 
 	/* Set custom log handler routines, so that GTK, GDK and GLIB never
