diff -rupN xdg_puppy-0.7.6-9-orig/jwm-xdgmenu/jwm-xdgmenu.c xdg_puppy-0.7.6-9/jwm-xdgmenu/jwm-xdgmenu.c
--- xdg_puppy-0.7.6-9-orig/jwm-xdgmenu/jwm-xdgmenu.c	2021-03-09 08:37:51.266137600 +0200
+++ xdg_puppy-0.7.6-9/jwm-xdgmenu/jwm-xdgmenu.c	2021-03-10 22:19:11.973052643 +0200
@@ -165,11 +165,28 @@ process_directory(GMenuTreeDirectory *di
 void
 process_entry(GMenuTreeEntry *entry)
 {
+  const char *tmp;
+  char *exec;
+  size_t len;
 
-  g_printf("<Program label=\"%s\" icon=\"%s\">%s</Program>\n",
+  tmp = gmenu_tree_entry_get_exec (entry);
+  exec = tmp;
+  len = strlen(exec);
+  if ((len > 3) && (exec[len - 3] == ' ') && (exec[len - 2] == '%'))
+  {
+    exec = g_strndup(exec, len - 3);
+  }
+
+  g_printf("<Program label=\"%s\" icon=\"%s\" tooltip=\"%s\">%s</Program>\n",
             gmenu_tree_entry_get_name(entry),
             gmenu_tree_entry_get_icon (entry),
-            gmenu_tree_entry_get_exec (entry));
+            gmenu_tree_entry_get_comment (entry),
+            exec);
+
+  if (exec != tmp)
+  {
+    g_free(exec);
+  }
 }
 
 /*=============================================================================
