diff -rupN xdg_puppy-0.7.6-9-orig/jwm-xdgmenu/jwm-xdgmenu.c xdg_puppy-0.7.6-9/jwm-xdgmenu/jwm-xdgmenu.c
--- xdg_puppy-0.7.6-9-orig/jwm-xdgmenu/jwm-xdgmenu.c	2021-10-01 18:39:19.718029746 +0300
+++ xdg_puppy-0.7.6-9/jwm-xdgmenu/jwm-xdgmenu.c	2021-10-01 18:42:19.617472367 +0300
@@ -165,11 +165,54 @@ process_directory(GMenuTreeDirectory *di
 void
 process_entry(GMenuTreeEntry *entry)
 {
+  const char *tmp, *name, *comment;
+  char *exec, *cmd;
+  size_t len;
 
-  g_printf("<Program label=\"%s\" icon=\"%s\">%s</Program>\n",
-            gmenu_tree_entry_get_name(entry),
-            gmenu_tree_entry_get_icon (entry),
-            gmenu_tree_entry_get_exec (entry));
+  tmp = gmenu_tree_entry_get_exec (entry);
+  exec = (char *)tmp;
+  len = strlen(exec);
+  if ((len > 3) && (exec[len - 3] == ' ') && (exec[len - 2] == '%'))
+  {
+    exec = g_strndup(exec, len - 3);
+  }
+
+  name = gmenu_tree_entry_get_name(entry);
+
+  if (gmenu_tree_entry_get_launch_in_terminal (entry))
+  {
+    cmd = g_strdup_printf("rxvt -title '%s' -e sh -c '%s'", name, exec);
+  }
+  else
+  {
+    cmd = exec;
+  }
+
+  comment = gmenu_tree_entry_get_comment (entry);
+  if (comment && comment[0])
+  {
+    g_printf("<Program label=\"%s\" icon=\"%s\" tooltip=\"%s\">%s</Program>\n",
+              name,
+              gmenu_tree_entry_get_icon (entry),
+              comment,
+              cmd);
+  }
+  else
+  {
+    g_printf("<Program label=\"%s\" icon=\"%s\">%s</Program>\n",
+              name,
+              gmenu_tree_entry_get_icon (entry),
+              cmd);
+  }
+
+  if (cmd != exec)
+  {
+    g_free(cmd);
+  }
+  if (exec != tmp)
+  {
+    g_free(exec);
+  }
 }
 
 /*=============================================================================
