diff -rupN xdg_puppy-0.7.6-9-orig/jwm-xdgmenu/jwm-xdgmenu.c xdg_puppy-0.7.6-9/jwm-xdgmenu/jwm-xdgmenu.c
--- xdg_puppy-0.7.6-9-orig/jwm-xdgmenu/jwm-xdgmenu.c	2021-03-10 22:39:55.078630581 +0200
+++ xdg_puppy-0.7.6-9/jwm-xdgmenu/jwm-xdgmenu.c	2021-03-14 21:39:50.935884915 +0200
@@ -165,11 +165,39 @@ process_directory(GMenuTreeDirectory *di
 void
 process_entry(GMenuTreeEntry *entry)
 {
+  const char *tmp, *comment;
+  char *exec;
+  size_t len;
 
-  g_printf("<Program label=\"%s\" icon=\"%s\">%s</Program>\n",
-            gmenu_tree_entry_get_name(entry),
-            gmenu_tree_entry_get_icon (entry),
-            gmenu_tree_entry_get_exec (entry));
+  tmp = gmenu_tree_entry_get_exec (entry);
+  exec = tmp;
+  len = strlen(exec);
+  if ((len > 3) && (exec[len - 3] == ' ') && (exec[len - 2] == '%'))
+  {
+    exec = g_strndup(exec, len - 3);
+  }
+
+  comment = gmenu_tree_entry_get_comment (entry);
+  if (comment && comment[0])
+  {
+    g_printf("<Program label=\"%s\" icon=\"%s\" tooltip=\"%s\">%s</Program>\n",
+              gmenu_tree_entry_get_name(entry),
+              gmenu_tree_entry_get_icon (entry),
+              comment,
+              exec);
+  }
+  else
+  {
+    g_printf("<Program label=\"%s\" icon=\"%s\">%s</Program>\n",
+              gmenu_tree_entry_get_name(entry),
+              gmenu_tree_entry_get_icon (entry),
+              exec);
+  }
+
+  if (exec != tmp)
+  {
+    g_free(exec);
+  }
 }
 
 /*=============================================================================
