diff -rupN xdg_puppy-0.7.6-9-orig/jwm-xdgmenu/jwm-xdgmenu.c xdg_puppy-0.7.6-9/jwm-xdgmenu/jwm-xdgmenu.c
--- xdg_puppy-0.7.6-9-orig/jwm-xdgmenu/jwm-xdgmenu.c	2022-04-25 16:34:20.875746887 +0800
+++ xdg_puppy-0.7.6-9/jwm-xdgmenu/jwm-xdgmenu.c	2022-04-25 16:38:08.935843086 +0800
@@ -212,6 +212,10 @@ process_entry(GMenuTreeEntry *entry)
   {
     exec = g_strndup(exec, len - 3);
   }
+  else if ((len > 4) && (exec[len - 2] == '@') && (exec[len - 1] == '@'))
+  {
+    exec = g_strndup(exec, strstr(exec, "@@") - exec);
+  }
 
   name = gmenu_tree_entry_get_name(entry);
 
