diff -rupN xdg_puppy-0.7.6-9-orig/jwm-xdgmenu/jwm-xdgmenu.c xdg_puppy-0.7.6-9/jwm-xdgmenu/jwm-xdgmenu.c
--- xdg_puppy-0.7.6-9-orig/jwm-xdgmenu/jwm-xdgmenu.c	2022-04-29 22:10:11.701935725 +0800
+++ xdg_puppy-0.7.6-9/jwm-xdgmenu/jwm-xdgmenu.c	2022-04-29 22:12:40.906103926 +0800
@@ -217,11 +217,25 @@ process_entry(GMenuTreeEntry *entry)
 
   if (gmenu_tree_entry_get_launch_in_terminal (entry))
   {
-    cmd = g_strdup_printf("rxvt -title '%s' -e sh -c '%s'", name, exec);
+    if (g_str_has_prefix(exec, "/usr/bin/flatpak run"))
+    {
+      cmd = g_strdup_printf("rxvt -title '%s' -e sh -c 'run-as-spot %s'", name, exec);
+    }
+    else
+    {
+      cmd = g_strdup_printf("rxvt -title '%s' -e sh -c '%s'", name, exec);
+    }
   }
   else
   {
-    cmd = exec;
+    if (g_str_has_prefix(exec, "/usr/bin/flatpak run"))
+    {
+      cmd = g_strdup_printf("run-as-spot %s", exec);
+    }
+    else
+    {
+      cmd = exec;
+    }
   }
 
   comment = gmenu_tree_entry_get_comment (entry);
