diff -rupN xdg_puppy-0.7.6-9-orig/jwm-xdgmenu/jwm-xdgmenu.c xdg_puppy-0.7.6-9/jwm-xdgmenu/jwm-xdgmenu.c
--- xdg_puppy-0.7.6-9-orig/jwm-xdgmenu/jwm-xdgmenu.c	2022-07-07 09:23:03.246665962 +0300
+++ xdg_puppy-0.7.6-9/jwm-xdgmenu/jwm-xdgmenu.c	2022-07-07 09:51:08.043483904 +0300
@@ -111,7 +111,7 @@ show_help()
 void
 process_directory(GMenuTreeDirectory *directory, char *menheight, GHashTable *history)
 {
-	int hasSeparator = 0, first = 1;
+	int hasSeparator = 0, first = 1, hadSeparator = 0;
 	char *mheight = menheight;
 	
   GMenuTreeItemType entryType;
@@ -155,21 +155,24 @@ start:
 		case GMENU_TREE_ITEM_DIRECTORY:
 		  if (hasSeparator)
 		  {
-				if (!first)
+				if (!first && !hadSeparator)
 				{
 					process_separator(GMENU_TREE_SEPARATOR(item));
+					hadSeparator = 1;
 				}
 				hasSeparator = 0;
 		  }
 			process_directory(GMENU_TREE_DIRECTORY(item),menheight,history);
 			first = 0;
+			hadSeparator = 0;
 			break;
 		case GMENU_TREE_ITEM_ENTRY:
 		  if (hasSeparator)
 		  {
-				if (!first)
+				if (!first && !hadSeparator)
 				{
 					process_separator(GMENU_TREE_SEPARATOR(item));
+					hadSeparator = 1;
 				}
 				hasSeparator = 0;
 		  }
@@ -179,6 +182,7 @@ start:
 			{
 				process_entry(entry);
 				first = 0;
+				hadSeparator = 0;
 				g_hash_table_insert(history, g_strdup(path), (gpointer)1);
 			}
 			break;
