diff -rupN xdg_puppy-0.7.6-9-orig/jwm-xdgmenu/jwm-xdgmenu.c xdg_puppy-0.7.6-9/jwm-xdgmenu/jwm-xdgmenu.c
--- xdg_puppy-0.7.6-9-orig/jwm-xdgmenu/jwm-xdgmenu.c	2021-02-24 09:26:15.392214890 +0200
+++ xdg_puppy-0.7.6-9/jwm-xdgmenu/jwm-xdgmenu.c	2021-02-24 09:28:01.540157679 +0200
@@ -108,20 +108,34 @@ show_help()
 void
 process_directory(GMenuTreeDirectory *directory, char *menheight)
 {
-	int hasSeparator = 0;
+	int hasSeparator = 0, first = 1;
 	char *mheight = menheight;
 	
-  g_printf("<Menu label=\"%s\" icon=\"%s\" height=\"%s\">\n",
-            gmenu_tree_directory_get_name(directory),
-            gmenu_tree_directory_get_icon(directory),
-            mheight);
-
   GMenuTreeItemType entryType;
 
   GSList *entryList = gmenu_tree_directory_get_contents (directory);
 
   GSList *l;
 
+  /* if the menu has no items, don't show an empty menu */
+  for (l = entryList; l; l = l->next)
+  {
+    GMenuTreeItem *item = l->data;
+
+    if (gmenu_tree_item_get_type (GMENU_TREE_ITEM(item)) == GMENU_TREE_ITEM_ENTRY)
+    {
+      goto start;
+    }
+  }
+
+  goto done;
+
+start:
+  g_printf("<Menu label=\"%s\" icon=\"%s\" height=\"%s\">\n",
+            gmenu_tree_directory_get_name(directory),
+            gmenu_tree_directory_get_icon(directory),
+            mheight);
+
   for (l = entryList; l; l = l->next)
   {
 
@@ -134,18 +148,26 @@ process_directory(GMenuTreeDirectory *di
 		case GMENU_TREE_ITEM_DIRECTORY:
 		  if (hasSeparator)
 		  {
-				process_separator(GMENU_TREE_SEPARATOR(item));
+				if (!first)
+				{
+					process_separator(GMENU_TREE_SEPARATOR(item));
+				}
 				hasSeparator = 0;
 		  }
 			process_directory(GMENU_TREE_DIRECTORY(item),menheight);
+			first = 0;
 			break;
 		case GMENU_TREE_ITEM_ENTRY:
 		  if (hasSeparator)
 		  {
-				process_separator(GMENU_TREE_SEPARATOR(item));
+				if (!first)
+				{
+					process_separator(GMENU_TREE_SEPARATOR(item));
+				}
 				hasSeparator = 0;
 		  }
 			process_entry(GMENU_TREE_ENTRY(item));
+			first = 0;
 			break;
 		case GMENU_TREE_ITEM_SEPARATOR:
 			hasSeparator = 1;
@@ -156,6 +178,7 @@ process_directory(GMenuTreeDirectory *di
 
   }
   g_printf("</Menu>\n");
+done:
   g_slist_free (entryList);
 }
 
