diff -rupN connman-gtk-b72c6ab3bb19c07325c8e659902b046daa23c506-orig/src/service.h connman-gtk-b72c6ab3bb19c07325c8e659902b046daa23c506/src/service.h
--- connman-gtk-b72c6ab3bb19c07325c8e659902b046daa23c506-orig/src/service.h	2021-12-24 16:40:19.410888498 +0200
+++ connman-gtk-b72c6ab3bb19c07325c8e659902b046daa23c506/src/service.h	2021-12-25 16:58:50.157093450 +0200
@@ -53,11 +53,16 @@ struct service {
 };
 
 #define SIGNAL_TO_ICON(type, strength) \
-        ((strength) > 80 ? ("network-" type "-signal-excellent-symbolic") : \
-        (strength) > 55 ? ("network-" type "-signal-good-symbolic") : \
-        (strength) > 30 ? ("network-" type "-signal-ok-symbolic") : \
-        (strength) > 5 ? ("network-" type "-signal-weak-symbolic") : \
-        ("network-" type "-signal-none-symbolic"))
+        ((strength) >= 90 ? ("qual_9") : \
+        (strength) >= 80 ? ("qual_8") : \
+        (strength) >= 70 ? ("qual_7") : \
+        (strength) >= 60 ? ("qual_6") : \
+        (strength) >= 50 ? ("qual_5") : \
+        (strength) >= 40 ? ("qual_4") : \
+        (strength) >= 30 ? ("qual_3") : \
+        (strength) >= 20 ? ("qual_2") : \
+        (strength) >= 10 ? ("qual_1") : \
+        ("qual_0"))
 
 struct service *service_create(struct technology *tech, GDBusProxy *proxy,
                                const gchar *path, GVariant *properties);
diff -rupN connman-gtk-b72c6ab3bb19c07325c8e659902b046daa23c506-orig/src/wireless.c connman-gtk-b72c6ab3bb19c07325c8e659902b046daa23c506/src/wireless.c
--- connman-gtk-b72c6ab3bb19c07325c8e659902b046daa23c506-orig/src/wireless.c	2021-12-24 16:40:19.414888498 +0200
+++ connman-gtk-b72c6ab3bb19c07325c8e659902b046daa23c506/src/wireless.c	2021-12-25 16:58:58.745093654 +0200
@@ -171,7 +171,7 @@ void service_wireless_init(struct servic
 
 	serv->data = item;
 	item->parent = serv;
-	item->favourite = gtk_image_new_from_icon_name("object-select-symbolic",
+	item->favourite = gtk_image_new_from_icon_name("emblem-default",
 						       GTK_ICON_SIZE_MENU);
 	item->security = gtk_image_new_from_icon_name("", GTK_ICON_SIZE_MENU);
 	item->signal = gtk_image_new_from_icon_name("", GTK_ICON_SIZE_MENU);
