diff -rupN connman-gtk-b72c6ab3bb19c07325c8e659902b046daa23c506-orig/src/configurator.c connman-gtk-b72c6ab3bb19c07325c8e659902b046daa23c506/src/configurator.c
--- connman-gtk-b72c6ab3bb19c07325c8e659902b046daa23c506-orig/src/configurator.c	2021-12-24 16:40:19.410888498 +0200
+++ connman-gtk-b72c6ab3bb19c07325c8e659902b046daa23c506/src/configurator.c	2021-12-24 16:40:26.998888342 +0200
@@ -111,6 +111,5 @@ void config_window_open(GtkApplication *
 #else
 
 void config_load(GtkApplication *app) {}
-void config_window_open(gpointer *ignored, gpointer user_data) {}
 
 #endif /* HAVE_CONFIG_SETTINGS */
