diff -rupN linux-3.1-orig//fs/aufs/loop.c linux-3.1//fs/aufs/loop.c
--- linux-3.1-orig//fs/aufs/loop.c	2011-11-08 15:54:01.828001000 +0200
+++ linux-3.1//fs/aufs/loop.c	2011-11-09 20:43:17.094074623 +0200
@@ -28,6 +28,9 @@
  */
 int au_test_loopback_overlap(struct super_block *sb, struct dentry *h_adding)
 {
+    /* allow Squashfs file systems to be used as Aufs branches */
+	return 0;
+	
 	struct super_block *h_sb;
 	struct loop_device *l;
 
