diff -rupN busybox-1.36.1-orig/networking/tc.c busybox-1.36.1/networking/tc.c
--- busybox-1.36.1-orig/networking/tc.c	2024-03-03 20:43:59.423846018 +0200
+++ busybox-1.36.1/networking/tc.c	2024-03-03 20:44:50.036173484 +0200
@@ -231,6 +231,7 @@ static int cbq_parse_opt(int argc, char
 	return 0;
 }
 #endif
+#if 0
 static int cbq_print_opt(struct rtattr *opt)
 {
 	struct rtattr *tb[TCA_CBQ_MAX+1];
@@ -322,6 +323,7 @@ static int cbq_print_opt(struct rtattr *
  done:
 	return 0;
 }
+#endif
 
 static FAST_FUNC int print_qdisc(
 		const struct sockaddr_nl *who UNUSED_PARAM,
@@ -372,8 +374,6 @@ static FAST_FUNC int print_qdisc(
 		int qqq = index_in_strings(_q_, name);
 		if (qqq == 0) { /* pfifo_fast aka prio */
 			prio_print_opt(tb[TCA_OPTIONS]);
-		} else if (qqq == 1) { /* class based queuing */
-			cbq_print_opt(tb[TCA_OPTIONS]);
 		} else {
 			/* don't know how to print options for this qdisc */
 			printf("(options for %s)", name);
@@ -442,9 +442,11 @@ static FAST_FUNC int print_class(
 		int qqq = index_in_strings(_q_, name);
 		if (qqq == 0) { /* pfifo_fast aka prio */
 			/* nothing. */ /*prio_print_opt(tb[TCA_OPTIONS]);*/
+#if 0
 		} else if (qqq == 1) { /* class based queuing */
 			/* cbq_print_copt() is identical to cbq_print_opt(). */
 			cbq_print_opt(tb[TCA_OPTIONS]);
+#endif
 		} else {
 			/* don't know how to print options for this class */
 			printf("(options for %s)", name);
