--- missingdefaultapp.orig	2022-12-12 16:54:17.287701945 +0530
+++ missingdefaultapp	2022-12-12 16:54:44.541757676 +0530
@@ -6,17 +6,17 @@
 /usr/lib/gtkdialog/box_yesno \
 --info \
 --image-icon info.svg \
---yes-label "Open PPM" \
+--yes-label "Open Synaptic" \
 --no-label "Open Default Apps" \
 --yes-icon package_pet.svg \
 --no-icon puppy_config.svg \
 "Configure default apps" \
 "Configure your default $1 by either installing one from PPM or openning Puppy Apps
 if an appropriate application is on your system." \
-"Click Open PPM to start Puppy Package Manager or Open Puppy Apps to configure."
+"Click Open Synaptic to start synaptic or Open Puppy Apps to configure."
 EXITCODE=$?
 echo $EXITCODE
 case $EXITCODE in
-	0)exec ppm;;
+	0)exec synaptic;;
 	1)exec puppyapps;;
 esac
