diff -rupN Viewnior-viewnior-1.8-orig/src/uni-utils.c Viewnior-viewnior-1.8/src/uni-utils.c
--- Viewnior-viewnior-1.8-orig/src/uni-utils.c	2023-01-13 14:55:15.956336191 +0200
+++ Viewnior-viewnior-1.8/src/uni-utils.c	2023-01-13 14:55:45.984452919 +0200
@@ -104,7 +104,7 @@ uni_rectangle_get_rects_around (GdkRecta
 VnrPrefsDesktop
 uni_detect_desktop_environment ()
 {
-    VnrPrefsDesktop environment = VNR_PREFS_DESKTOP_GNOME3;
+    VnrPrefsDesktop environment = VNR_PREFS_DESKTOP_PUPPY;
 
     gchar *xdg_current_desktop = g_ascii_strup(g_getenv("XDG_CURRENT_DESKTOP"), -1);
     gchar *xdg_session_desktop = g_ascii_strup(g_getenv("XDG_SESSION_DESKTOP"), -1);
@@ -140,7 +140,7 @@ uni_detect_desktop_environment ()
     }
     else
     {
-        g_warning("Cannot detect desktop environment. Defaulting to GNOME 3.\n");
+        g_warning("Cannot detect desktop environment. Defaulting to Puppy.\n");
     }
 
     g_free(xdg_current_desktop);
