diff -Naur busybox-1.24.2/miscutils/timeout.c busybox-1.24.2-patched/miscutils/timeout.c
--- busybox-1.24.2/miscutils/timeout.c	2016-02-11 11:50:37.000000000 +0000
+++ busybox-1.24.2-patched/miscutils/timeout.c	2016-05-22 17:53:29.088894730 +0000
@@ -28,6 +28,14 @@
  * rewrite  14-11-2008 vda
  */
 
+//                                   alias   _main()                                help (xx_xx_usage)
+//applet:IF_TIMEOUT( APPLET_ODDNAME(waitmax, timeout, BB_DIR_USR_BIN, BB_SUID_DROP, waitmax) )
+
+//usage:#define waitmax_trivial_usage
+//usage:       "[-s SIG] MAXTIME PROG ARGS"
+//usage:#define waitmax_full_usage "\n\n"
+//usage:       "Runs PROG. Sends SIG to it if it is not gone in MAXTIME seconds.\n"
+
 //usage:#define timeout_trivial_usage
 //usage:       "[-t SECS] [-s SIG] PROG ARGS"
 //usage:#define timeout_full_usage "\n\n"
@@ -36,6 +44,8 @@
 
 #include "libbb.h"
 
+#define WAITMAX (ENABLE_TIMEOUT && applet_name[0] == 'w')
+
 int timeout_main(int argc, char **argv) MAIN_EXTERNALLY_VISIBLE;
 int timeout_main(int argc UNUSED_PARAM, char **argv)
 {
@@ -48,14 +58,22 @@
 	char *sv1, *sv2;
 #endif
 	const char *opt_s = "TERM";
-
 	/* -p option is not documented, it is needed to support NOMMU. */
-
-	/* -t SECONDS; -p PARENT_PID */
-	opt_complementary = "t+" USE_FOR_NOMMU(":p+");
-	/* '+': stop at first non-option */
-	getopt32(argv, "+s:t:" USE_FOR_NOMMU("p:"), &opt_s, &timeout, &parent);
-	/*argv += optind; - no, wait for bb_daemonize_or_rexec! */
+	if (WAITMAX) {
+		// Need at least two arguments
+		opt_complementary = "-2:" USE_FOR_NOMMU(":p+");
+		getopt32(argv, "s:" USE_FOR_NOMMU("p:"), &opt_s, &parent);
+		timeout = strtol(argv[optind], NULL, 10);
+		if (!timeout) bb_show_usage();
+		optind++;
+	}
+	else {
+		/* -t SECONDS; -p PARENT_PID */
+		opt_complementary = "t+" USE_FOR_NOMMU(":p+");
+		/* '+': stop at first non-option */
+		getopt32(argv, "+s:t:" USE_FOR_NOMMU("p:"), &opt_s, &timeout, &parent);
+		/*argv += optind; - no, wait for bb_daemonize_or_rexec! */
+	}
 	signo = get_signum(opt_s);
 	if (signo < 0)
 		bb_error_msg_and_die("unknown signal '%s'", opt_s);
