diff -Naur busybox-1.25.0/include/libbb.h busybox-1.25.0-patched/include/libbb.h
--- busybox-1.25.0/include/libbb.h	2016-06-19 13:16:27.000000000 +0000
+++ busybox-1.25.0-patched/include/libbb.h	2016-06-22 13:37:10.000000000 +0000
@@ -43,6 +43,7 @@
 #include <sys/mman.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
+#include <sys/sysinfo.h>
 #include <sys/time.h>
 #include <sys/types.h>
 #ifndef major
diff -Naur busybox-1.25.0/include/platform.h busybox-1.25.0-patched/include/platform.h
--- busybox-1.25.0/include/platform.h	2015-12-30 15:01:53.000000000 +0000
+++ busybox-1.25.0-patched/include/platform.h	2016-06-22 13:37:10.000000000 +0000
@@ -501,6 +501,13 @@
 # undef HAVE_NET_ETHERNET_H
 #endif
 
+#if defined(__musl__)
+# undef HAVE_SETBIT
+# include <stddef.h>
+# include <termios.h>
+# include <sys/ioctl.h>
+#endif
+
 /*
  * Now, define prototypes for all the functions defined in platform.c
  * These must come after all the HAVE_* macros are defined (or not)
