diff -Naur busybox-1.26.1/shell/ash.c busybox-1.26.1-patched/shell/ash.c
--- busybox-1.26.1/shell/ash.c	2017-01-01 20:01:05.000000000 +0800
+++ busybox-1.26.1-patched/shell/ash.c	2017-01-06 10:08:45.329095750 +0800
@@ -1284,6 +1284,8 @@
 {
 	va_list ap;
 
+	exitstatus = 2;
+
 	va_start(ap, msg);
 	ash_vmsg_and_raise(EXERROR, msg, ap);
 	/* NOTREACHED */
@@ -9588,11 +9590,12 @@
 	}
 
 	if (status) {
+ bail:
+		exitstatus = status;
+
 		/* We have a redirection error. */
 		if (spclbltin > 0)
 			raise_exception(EXERROR);
- bail:
-		exitstatus = status;
 		goto out;
 	}
 
diff -Naur busybox-1.26.1/shell/ash_test/ash-misc/exitcode2.tests busybox-1.26.1-patched/shell/ash_test/ash-misc/exitcode2.tests
--- busybox-1.26.1/shell/ash_test/ash-misc/exitcode2.tests	1970-01-01 08:00:00.000000000 +0800
+++ busybox-1.26.1-patched/shell/ash_test/ash-misc/exitcode2.tests	2017-01-06 10:07:03.099093783 +0800
@@ -0,0 +1,13 @@
+# syntax error should return status 2
+cat >test.sh <<EOF
+)
+EOF
+chmod +x test.sh
+./test.sh
+echo Done:$?
+rm -f test.sh
+
+# redirection error with special builtin should return status 1
+rm -f testfile
+(eval cat <testfile)
+echo Done:$?
