diff -Naur busybox-1.30.0/miscutils/bc.c busybox-1.30.0-patched/miscutils/bc.c
--- busybox-1.30.0/miscutils/bc.c	2018-12-30 23:14:20.000000000 +0800
+++ busybox-1.30.0-patched/miscutils/bc.c	2019-01-15 08:10:15.426665000 +0800
@@ -7373,7 +7373,11 @@
 
 	while (*argv) {
 		noscript = 0;
-		bc_vec_push(&G.files, argv++);
+		//if (access(*argv, F_OK) == 0)
+		//	zxc_vm_file(xfopen_for_read(*argv));
+		//else
+			zxc_vm_process(*argv);
+		argv++;
 	}
 
 	option_mask32 |= noscript; // set BC_FLAG_I if we need to interpret stdin
