diff -Naur busybox-1.25.0/include/libbb.h busybox-1.25.0-patched/include/libbb.h
--- busybox-1.25.0/include/libbb.h	2016-06-19 13:16:27.000000000 +0000
+++ busybox-1.25.0-patched/include/libbb.h	2016-06-22 13:37:10.000000000 +0000
@@ -43,6 +43,7 @@
 #include <sys/mman.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
+#include <sys/sysinfo.h>
 #include <sys/time.h>
 #include <sys/types.h>
 #ifndef major
diff -Naur busybox-1.25.0/include/platform.h busybox-1.25.0-patched/include/platform.h
--- busybox-1.25.0/include/platform.h	2015-12-30 15:01:53.000000000 +0000
+++ busybox-1.25.0-patched/include/platform.h	2016-06-22 13:37:10.000000000 +0000
@@ -501,6 +501,13 @@
 # undef HAVE_NET_ETHERNET_H
 #endif
 
+#if defined(__musl__)
+# undef HAVE_SETBIT
+# include <stddef.h>
+# include <termios.h>
+# include <sys/ioctl.h>
+#endif
+
 /*
  * Now, define prototypes for all the functions defined in platform.c
  * These must come after all the HAVE_* macros are defined (or not)
diff -Naur busybox-1.25.0/networking/tcpudp.c busybox-1.25.0-patched/networking/tcpudp.c
--- busybox-1.25.0/networking/tcpudp.c	2016-05-26 17:42:44.000000000 +0000
+++ busybox-1.25.0-patched/networking/tcpudp.c	2016-06-22 13:37:10.000000000 +0000
@@ -72,7 +72,7 @@
 /* Wants <limits.h> etc, thus included after libbb.h: */
 #ifdef __linux__
 #include <linux/types.h> /* for __be32 etc */
-#include <linux/netfilter_ipv4.h>
+//#include <linux/netfilter_ipv4.h>
 #endif
 
 // TODO: move into this file:
diff -Naur busybox-1.25.0/util-linux/fdisk.c busybox-1.25.0-patched/util-linux/fdisk.c
--- busybox-1.25.0/util-linux/fdisk.c	2015-10-11 13:31:51.000000000 +0000
+++ busybox-1.25.0-patched/util-linux/fdisk.c	2016-06-22 13:37:10.000000000 +0000
@@ -550,7 +550,7 @@
 {
 #if ENABLE_FDISK_SUPPORT_LARGE_DISKS
 	off64_t off = (off64_t)secno * sector_size;
-	if (lseek64(dev_fd, off, SEEK_SET) == (off64_t) -1)
+	if (lseek(dev_fd, off, SEEK_SET) == (off64_t) -1)
 		fdisk_fatal(unable_to_seek);
 #else
 	uint64_t off = (uint64_t)secno * sector_size;
