diff -rupN pcmanfm-1.3.2-orig/data/pcmanfm.conf pcmanfm-1.3.2/data/pcmanfm.conf
--- pcmanfm-1.3.2-orig/data/pcmanfm.conf	2022-03-28 19:43:22.688693759 +0300
+++ pcmanfm-1.3.2/data/pcmanfm.conf	2022-03-29 10:11:14.452689377 +0300
@@ -15,7 +15,7 @@ show_wm_menu=0
 
 [ui]
 win_width=640
-win_height=480
+win_height=240
 splitter_pos=150
 side_pane_mode=1
 view_mode=0
@@ -23,3 +23,6 @@ show_hidden=0
 sort_type=0
 sort_by=2
 max_tab_chars=32
+show_statusbar=0
+pathbar_mode_buttons=1
+desktop_folder_new_win=1
