diff -rupN pcmanfm-1.3.2-orig/data/pcmanfm.conf pcmanfm-1.3.2/data/pcmanfm.conf
--- pcmanfm-1.3.2-orig/data/pcmanfm.conf	2022-03-28 19:43:22.688693759 +0300
+++ pcmanfm-1.3.2/data/pcmanfm.conf	2022-03-29 10:44:58.828953759 +0300
@@ -7,15 +7,18 @@ mount_removable=1
 autorun=1
 
 [desktop]
-wallpaper_mode=0
+wallpaper_mode=stretch
+wallpaper=/usr/share/backgrounds/default.svg
+desktop_font=Sans 10
 desktop_bg=#000000
 desktop_fg=#ffffff
 desktop_shadow=#000000
 show_wm_menu=0
+show_mounts=1
 
 [ui]
 win_width=640
-win_height=480
+win_height=240
 splitter_pos=150
 side_pane_mode=1
 view_mode=0
@@ -23,3 +26,6 @@ show_hidden=0
 sort_type=0
 sort_by=2
 max_tab_chars=32
+show_statusbar=0
+pathbar_mode_buttons=1
+desktop_folder_new_win=1
