diff -rupN pcmanfm-1.3.2-orig/data/pcmanfm.conf pcmanfm-1.3.2/data/pcmanfm.conf
--- pcmanfm-1.3.2-orig/data/pcmanfm.conf	2022-01-08 18:08:39.678346789 +0800
+++ pcmanfm-1.3.2/data/pcmanfm.conf	2022-01-08 18:27:26.386310429 +0800
@@ -15,11 +15,13 @@ show_wm_menu=0
 
 [ui]
 win_width=640
-win_height=480
+win_height=240
 splitter_pos=150
-side_pane_mode=1
+side_pane_mode=hidden;places
 view_mode=0
 show_hidden=0
 sort_type=0
 sort_by=2
 max_tab_chars=32
+show_statusbar=0
+pathbar_mode_buttons=1
