diff -rupN pcmanfm-1.3.2-orig/src/main-win.c pcmanfm-1.3.2/src/main-win.c
--- pcmanfm-1.3.2-orig/src/main-win.c	2021-12-24 13:27:41.707090425 +0200
+++ pcmanfm-1.3.2/src/main-win.c	2021-12-24 13:28:24.079089555 +0200
@@ -955,7 +955,7 @@ static void fm_main_win_init(FmMainWin *
     /* the location bar */
     win->location = fm_path_entry_new();
     g_signal_connect(win->location, "activate", G_CALLBACK(on_location_activate), win);
-    if(geteuid() == 0) /* if we're using root, Give the user some warnings */
+    if(0) /* if we're using root, Give the user some warnings */
     {
         GtkWidget* warning = gtk_image_new_from_stock(GTK_STOCK_DIALOG_WARNING, GTK_ICON_SIZE_SMALL_TOOLBAR);
         gtk_widget_set_tooltip_markup(warning, _("You are in super user mode"));
