diff -rupN swaylock-1.5-orig/shadow.c swaylock-1.5/shadow.c
--- swaylock-1.5-orig/shadow.c	2022-01-01 10:37:29.459876775 +0800
+++ swaylock-1.5/shadow.c	2022-01-01 10:37:48.223876170 +0800
@@ -32,11 +32,13 @@ void initialize_pw_backend(int argc, cha
 		swaylock_log_errno(LOG_ERROR, "Unable to drop root");
 		exit(EXIT_FAILURE);
 	}
+#if 0
 	if (setuid(0) != -1) {
 		swaylock_log_errno(LOG_ERROR, "Unable to drop root (we shouldn't be "
 			"able to restore it after setuid)");
 		exit(EXIT_FAILURE);
 	}
+#endif
 }
 
 void run_pw_backend_child(void) {
@@ -64,9 +66,11 @@ void run_pw_backend_child(void) {
 	if (setuid(getuid()) != 0) {
 		exit(EXIT_FAILURE);
 	}
+#if 0
 	if (setuid(0) != -1) {
 		exit(EXIT_FAILURE);
 	}
+#endif
 
 	/* This code does not run as root */
 	swaylock_log(LOG_DEBUG, "Prepared to authorize user %s", pwent->pw_name);
