diff -rup geany-1.37.1-orig/src/keyfile.c geany-1.37.1/src/keyfile.c
--- geany-1.37.1-orig/src/keyfile.c	2020-11-08 19:14:52.000000000 +0200
+++ geany-1.37.1/src/keyfile.c	2021-01-01 13:17:00.564197457 +0200
@@ -1022,8 +1022,8 @@ static void load_ui_prefs(GKeyFile *conf
 	gint *geo;
 	gsize geo_len;
 
-	ui_prefs.sidebar_visible = utils_get_setting_boolean(config, PACKAGE, "sidebar_visible", TRUE);
-	ui_prefs.msgwindow_visible = utils_get_setting_boolean(config, PACKAGE, "msgwindow_visible", TRUE);
+	ui_prefs.sidebar_visible = utils_get_setting_boolean(config, PACKAGE, "sidebar_visible", FALSE);
+	ui_prefs.msgwindow_visible = utils_get_setting_boolean(config, PACKAGE, "msgwindow_visible", FALSE);
 	ui_prefs.fullscreen = utils_get_setting_boolean(config, PACKAGE, "fullscreen", FALSE);
 	ui_prefs.custom_date_format = utils_get_setting_string(config, PACKAGE, "custom_date_format", "");
 	ui_prefs.custom_commands = g_key_file_get_string_list(config, PACKAGE, "custom_commands", NULL, NULL);
