diff -rup geany-1.37.1-orig/src/keyfile.c geany-1.37.1/src/keyfile.c
--- geany-1.37.1-orig/src/keyfile.c	2021-02-06 09:25:02.886925427 +0200
+++ geany-1.37.1/src/keyfile.c	2021-02-06 09:26:46.926329908 +0200
@@ -79,7 +79,7 @@
 #elif defined(__APPLE__)
 #define GEANY_DEFAULT_TOOLS_TERMINAL	"open -a terminal %c"
 #else
-#define GEANY_DEFAULT_TOOLS_TERMINAL	"xterm -e \"/bin/sh %c\""
+#define GEANY_DEFAULT_TOOLS_TERMINAL	"urxvt -e /bin/sh %c"
 #endif
 #ifdef __APPLE__
 #define GEANY_DEFAULT_TOOLS_BROWSER		"open -a safari"
@@ -87,7 +87,7 @@
 #define GEANY_DEFAULT_FONT_MSG_WINDOW	"Menlo Medium 12"
 #define GEANY_DEFAULT_FONT_EDITOR		"Menlo Medium 12"
 #else
-#define GEANY_DEFAULT_TOOLS_BROWSER		"firefox"
+#define GEANY_DEFAULT_TOOLS_BROWSER		"defaultbrowser"
 #define GEANY_DEFAULT_FONT_SYMBOL_LIST	"Sans 9"
 #define GEANY_DEFAULT_FONT_MSG_WINDOW	"Monospace 9"
 #define GEANY_DEFAULT_FONT_EDITOR		"Monospace 10"
@@ -1022,8 +1022,8 @@ static void load_ui_prefs(GKeyFile *conf
 	gint *geo;
 	gsize geo_len;
 
-	ui_prefs.sidebar_visible = utils_get_setting_boolean(config, PACKAGE, "sidebar_visible", TRUE);
-	ui_prefs.msgwindow_visible = utils_get_setting_boolean(config, PACKAGE, "msgwindow_visible", TRUE);
+	ui_prefs.sidebar_visible = utils_get_setting_boolean(config, PACKAGE, "sidebar_visible", FALSE);
+	ui_prefs.msgwindow_visible = utils_get_setting_boolean(config, PACKAGE, "msgwindow_visible", FALSE);
 	ui_prefs.fullscreen = utils_get_setting_boolean(config, PACKAGE, "fullscreen", FALSE);
 	ui_prefs.custom_date_format = utils_get_setting_string(config, PACKAGE, "custom_date_format", "");
 	ui_prefs.custom_commands = g_key_file_get_string_list(config, PACKAGE, "custom_commands", NULL, NULL);
