diff -rup pa-applet-97c18dd8a30e65c4a1d27d3d06248e60f1b07087-orig/configure.ac pa-applet-97c18dd8a30e65c4a1d27d3d06248e60f1b07087/configure.ac
--- pa-applet-97c18dd8a30e65c4a1d27d3d06248e60f1b07087-orig/configure.ac	2021-02-04 16:22:19.995251830 +0200
+++ pa-applet-97c18dd8a30e65c4a1d27d3d06248e60f1b07087/configure.ac	2021-02-04 16:22:34.103326098 +0200
@@ -15,7 +15,6 @@ PKG_CHECK_MODULES([GLIB], [glib-2.0])
 PKG_CHECK_MODULES([GTK3], [gtk+-3.0])
 PKG_CHECK_MODULES([LIBPULSE], [libpulse])
 PKG_CHECK_MODULES([LIBPULSE_GLIB], [libpulse-mainloop-glib])
-PKG_CHECK_MODULES([LIBNOTIFY], [libnotify])
 PKG_CHECK_MODULES([XLIB], [x11])
 
 AC_CONFIG_FILES([Makefile man/Makefile src/Makefile])
diff -rup pa-applet-97c18dd8a30e65c4a1d27d3d06248e60f1b07087-orig/src/notifications.c pa-applet-97c18dd8a30e65c4a1d27d3d06248e60f1b07087/src/notifications.c
--- pa-applet-97c18dd8a30e65c4a1d27d3d06248e60f1b07087-orig/src/notifications.c	2021-02-04 16:22:19.995251830 +0200
+++ pa-applet-97c18dd8a30e65c4a1d27d3d06248e60f1b07087/src/notifications.c	2021-02-04 16:23:25.283595313 +0200
@@ -7,17 +7,14 @@
  *
  */
 
-#include <libnotify/notify.h>
 
 #include "audio_status.h"
 
 #define PROGRAM_NAME "pa-applet"
 
-gboolean have_notifications = FALSE;
-NotifyNotification *notification = NULL;
-
 void notifications_init(void)
 {
+#if 0
     if (notify_init(PROGRAM_NAME)) {
         // Create and configure the notification
         have_notifications = TRUE;
@@ -37,19 +34,23 @@ void notifications_init(void)
     else {
         g_printerr("Failed to initialize notifications\n");
     }
+#endif
 }
 
 void notifications_destroy(void)
 {
+#if 0
     if (have_notifications) {
         notify_uninit();
         if (notification)
             g_object_unref(G_OBJECT(notification));
     }
+#endif
 }
 
 void notifications_flash(void)
 {
+#if 0
     // Nothing to do if we don't support notifications
     if (!have_notifications || !notification)
         return;
@@ -77,4 +78,5 @@ void notifications_flash(void)
 
     // Show the notification
     notify_notification_show(notification, NULL);
+#endif
 }
