diff -rupN lxterminal-0.4.0-orig/src/setting.c lxterminal-0.4.0/src/setting.c
--- lxterminal-0.4.0-orig/src/setting.c	2021-03-12 20:20:26.156393104 +0200
+++ lxterminal-0.4.0/src/setting.c	2021-03-12 20:21:08.948864320 +0200
@@ -38,8 +38,8 @@ Setting * setting;
 ColorPreset color_presets[] = {
     {
         .name = "VGA",
-        .background_color = "#000000",
-        .foreground_color = "#aaaaaa",
+        .background_color = "#1a1a1a",
+        .foreground_color = "#f8f8f8",
         .palette = {
             "#000000", "#aa0000", "#00aa00", "#aa5500",
             "#0000aa", "#aa00aa", "#00aaaa", "#aaaaaa",
