diff -rup lxterminal-0.4.0-orig/src/setting.c lxterminal-0.4.0/src/setting.c
--- lxterminal-0.4.0-orig/src/setting.c	2024-01-26 11:00:15.817511151 +0530
+++ lxterminal-0.4.0/src/setting.c	2024-01-26 11:46:02.729178961 +0530
@@ -42,7 +42,7 @@ ColorPreset color_presets[] = {
         .foreground_color = "#aaaaaa",
         .palette = {
             "#000000", "#aa0000", "#00aa00", "#aa5500",
-            "#0000aa", "#aa00aa", "#00aaaa", "#aaaaaa",
+            "#001fff", "#aa00aa", "#00aaaa", "#aaaaaa",
             "#555555", "#ff5555", "#55ff55", "#ffff55",
             "#5555ff", "#ff55ff", "#55ffff", "#ffffff"
         }
@@ -350,10 +350,12 @@ Setting * load_setting()
     /* Initialize nonzero integer values to defaults. */
 #if VTE_CHECK_VERSION (0, 38, 0)
     setting->background_color.alpha = setting->foreground_color.alpha = 1;
-    setting->foreground_color.red = setting->foreground_color.green = setting->foreground_color.blue = (gdouble) 170/255;
+    setting->background_color.red = setting->background_color.green = setting->background_color.blue = (gdouble) 0x1a/255;
+    setting->foreground_color.red = setting->foreground_color.green = setting->foreground_color.blue = (gdouble) 0xf8/255;
 #else
     setting->background_alpha = 65535;
-    setting->foreground_color.red = setting->foreground_color.green = setting->foreground_color.blue = 0xaaaa;
+    setting->background_color.red = setting->background_color.green = setting->background_color.blue = 0x1a1a;
+    setting->foreground_color.red = setting->foreground_color.green = setting->foreground_color.blue = 0xf8f8;
 #endif
 
     /* Load configuration. */
