diff -rup lxterminal-0.4.0-orig/data/lxterminal.conf lxterminal-0.4.0/data/lxterminal.conf
--- lxterminal-0.4.0-orig/data/lxterminal.conf	2024-01-27 16:42:04.048250277 +0530
+++ lxterminal-0.4.0/data/lxterminal.conf	2024-01-28 03:03:44.136433337 +0530
@@ -2,3 +2,6 @@
 fontname=Monospace 10
 selchars=-A-Za-z0-9,./?%&#:_
 scrollback=1000
+bgcolor=rgb(36,31,49)
+fgcolor=rgb(222,221,218)
+boldbright=true
diff -rup lxterminal-0.4.0-orig/data/lxterminal.conf.in lxterminal-0.4.0/data/lxterminal.conf.in
--- lxterminal-0.4.0-orig/data/lxterminal.conf.in	2024-01-27 16:42:04.048250277 +0530
+++ lxterminal-0.4.0/data/lxterminal.conf.in	2024-01-28 03:05:37.924989756 +0530
@@ -2,3 +2,6 @@
 fontname=Monospace 10
 selchars=-A-Za-z0-9,./?%&#:_
 scrollback=1000
+bgcolor=rgb(36,31,49)
+fgcolor=rgb(222,221,218)
+boldbright=true
diff -rup lxterminal-0.4.0-orig/src/setting.c lxterminal-0.4.0/src/setting.c
--- lxterminal-0.4.0-orig/src/setting.c	2024-01-27 16:42:04.040251209 +0530
+++ lxterminal-0.4.0/src/setting.c	2024-01-28 03:18:46.139740603 +0530
@@ -37,6 +37,18 @@ Setting * setting;
 
 ColorPreset color_presets[] = {
     {
+        // add Puppy preset here; other modified settings in ../data/lxterminal.conf
+        .name = "Puppy",
+        .background_color = "#241f31",
+        .foreground_color = "#deddda",
+        .palette = {
+            "#000000", "#ff0000","#2ec27e", "#ff7600",
+            "#1c71d8", "#c061cb", "#00aaaa", "#deddda",
+            "#797979", "#ff5f5f", "#57e389", "#f6d32d",
+            "#62a0ea", "#ff55ff", "#55ffff", "#f6f5f4"
+        }
+    },
+    {
         .name = "VGA",
         .background_color = "#000000",
         .foreground_color = "#aaaaaa",
