diff -rup lxterminal-0.4.0-orig/src/setting.c lxterminal-0.4.0/src/setting.c
--- lxterminal-0.4.0-orig/src/setting.c	2024-01-27 16:42:04.040251209 +0530
+++ lxterminal-0.4.0/src/setting.c	2024-01-27 17:15:16.041523334 +0530
@@ -37,6 +37,17 @@ Setting * setting;
 
 ColorPreset color_presets[] = {
     {
+        .name = "Puppy",
+        .background_color = "#241f31",
+        .foreground_color = "#f8f8f8",
+        .palette = {
+            "#000000", "#ff0000", "#26a269", "#986a44",
+            "#1a5fb4", "#aa00aa", "#00aaaa", "#aaaaaa",
+            "#3d3846", "#ff5454", "#33d17a", "#f6d32d",
+            "#62a0ea", "#ff55ff", "#55ffff", "#ffffff"
+        }
+    },
+    {
         .name = "VGA",
         .background_color = "#000000",
         .foreground_color = "#aaaaaa",
