diff -rupN lxterminal-0.4.0-orig/src/setting.c lxterminal-0.4.0/src/setting.c
--- lxterminal-0.4.0-orig/src/setting.c	2021-03-12 20:20:26.156393104 +0200
+++ lxterminal-0.4.0/src/setting.c	2021-03-12 20:59:23.801655377 +0200
@@ -442,7 +442,11 @@ color_preset_does_not_exist:
         }
         g_clear_error(&error);
         setting->hide_scroll_bar = g_key_file_get_boolean(setting->keyfile, GENERAL_GROUP, HIDE_SCROLLBAR, NULL);
-        setting->hide_menu_bar = g_key_file_get_boolean(setting->keyfile, GENERAL_GROUP, HIDE_MENUBAR, NULL);
+        setting->hide_menu_bar = g_key_file_get_boolean(setting->keyfile, GENERAL_GROUP, HIDE_MENUBAR, &error);
+        if (error && (error->code == G_KEY_FILE_ERROR_KEY_NOT_FOUND)) {
+            setting->hide_menu_bar = TRUE;
+        }
+        g_clear_error(&error);
         setting->hide_close_button = g_key_file_get_boolean(setting->keyfile, GENERAL_GROUP, HIDE_CLOSE_BUTTON, NULL);
         setting->hide_pointer = g_key_file_get_boolean(setting->keyfile, GENERAL_GROUP, HIDE_POINTER, NULL);
         setting->word_selection_characters = g_key_file_get_string(setting->keyfile, GENERAL_GROUP, SEL_CHARS, NULL);
