diff -rup '--exclude=.git' '--exclude=.gitmodules' deadbeef-orig/configure.ac deadbeef/configure.ac
--- deadbeef-orig/configure.ac	2023-06-14 11:01:10.134752040 +0300
+++ deadbeef/configure.ac	2023-06-14 10:29:11.837368402 +0300
@@ -79,6 +79,7 @@ AC_SUBST(INSANE_CXXFLAGS)
 AC_ARG_ENABLE(nullout,  [AS_HELP_STRING([--disable-nullout ], [disable NULL output plugin (default: enabled)])], [enable_nullout=$enableval], [enable_nullout=yes])
 AC_ARG_ENABLE(alsa,     [AS_HELP_STRING([--disable-alsa    ], [disable ALSA output plugin (default: enabled)])], [enable_alsa=$enableval], [enable_alsa=yes])
 AC_ARG_ENABLE(oss,      [AS_HELP_STRING([--disable-oss     ], [disable Open Sound System output plugin (default: enabled)])], [enable_oss=$enableval], [enable_oss=yes])
+AC_ARG_ENABLE(pipewire, [AS_HELP_STRING([--disable-pipewire   ], [disable Pipewire output plugin (default: enabled)])], [enable_pipewire=$enableval], [enable_pipewire=yes])
 AC_ARG_ENABLE(pulse,    [AS_HELP_STRING([--disable-pulse   ], [disable PulseAudio output plugin (default: enabled)])], [enable_pulse=$enableval], [enable_pulse=yes])
 AC_ARG_ENABLE(coreaudio,[AS_HELP_STRING([--disable-coreaudio], [disable CoreAudio output plugin (default: enabled)])], [enable_coreaudio=$enableval], [enable_coreaudio=yes])
 AC_ARG_ENABLE(gtk3,     [AS_HELP_STRING([--disable-gtk3     ], [disable GTK3 version of gtkui plugin (default: enabled)])], [enable_gtk3=$enableval], [enable_gtk3=yes])
@@ -320,13 +321,25 @@ AS_IF([test "${enable_pulse}" != "no"],
         HAVE_PULSE=yes
         PULSE_DEPS_LIBS="-lpulse-simple"
         PULSE_DEPS_CFLAGS="-I../../$LIB/include/"
-        AC_SUBST(DBUS_DEPS_CFLAGS)
-        AC_SUBST(DBUS_DEPS_LIBS)
+        AC_SUBST(PULSE_DEPS_CFLAGS)
+        AC_SUBST(PULSE_DEPS_LIBS)
     ], [
         PKG_CHECK_MODULES(PULSE_DEPS, libpulse-simple, HAVE_PULSE=yes, HAVE_PULSE=no)
     ])
 ])
 
+AS_IF([test "${enable_pipewire}" != "no"], [
+    AS_IF([test "${enable_staticlink}" != "no"], [
+        HAVE_PIPEWIRE=yes
+        PIPEWIRE_DEPS_LIBS="-lpipewire-0.3"
+        PIPEWIRE_DEPS_CFLAGS="-D_REENTRANT -I../$LIB/include/pipewire-0.3 -I../$LIB/include/spa-0.2"
+        AC_SUBST(PIPEWIRE_DEPS_CFLAGS)
+        AC_SUBST(PIPEWIRE_DEPS_LIBS)
+    ], [
+        PKG_CHECK_MODULES(PIPEWIRE_DEPS, libpipewire-0.3, HAVE_PIPEWIRE=yes, HAVE_PIPEWIRE=no)
+    ])
+])
+
 AC_CHECK_HEADER([iconv.h],[],[iconv.h not found.])
 
 AC_CHECK_LIB([iconv], [main], [have_iconv=yes], [have_iconv=no])
@@ -619,6 +632,12 @@ AS_IF([test "${enable_pulseaudio}" != "n
     ])
 ])
 
+AS_IF([test "${enable_pipewire}" != "no"], [
+    AS_IF([test "${HAVE_PIPEWIRE}" = "yes"], [
+        HAVE_PIPEWIRE_PLUGIN=yes
+    ])
+])
+
 dnl lastfm plugin
 AS_IF([test "${enable_lfm}" != "no"], [
     AS_IF([test "${HAVE_CURL}" = "yes" -a "${enable_lastfm}" != "no"], [
@@ -854,6 +873,7 @@ AM_CONDITIONAL(HAVE_GTK3, test "x$HAVE_G
 AM_CONDITIONAL(HAVE_ALSA, test "x$HAVE_ALSAPLUGIN" = "xyes")
 AM_CONDITIONAL(HAVE_COREAUDIO, test "x$HAVE_COREAUDIO" = "xyes")
 AM_CONDITIONAL(HAVE_FFMPEG, test "x$HAVE_FFMPEGPLUGIN" = "xyes")
+AM_CONDITIONAL(HAVE_PIPEWIRE, test "x$HAVE_PIPEWIRE_PLUGIN" = "xyes")
 AM_CONDITIONAL(HAVE_PULSE, test "x$HAVE_PULSEPLUGIN" = "xyes")
 AM_CONDITIONAL(HAVE_ARTWORK, test "x$HAVE_ARTWORK" = "xyes")
 AM_CONDITIONAL(ARTWORK_USE_VFS_CURL, test "x$ARTWORK_USE_VFS_CURL" = "xyes")
@@ -1027,6 +1047,7 @@ PRINT_PLUGIN_INFO([opus],[ogg opus playe
 PRINT_PLUGIN_INFO([oss],[oss output plugin],[test "x$HAVE_OSS" = "xyes"])
 PRINT_PLUGIN_INFO([pltbrowser],[playlist browser gui plugin],[test "x$HAVE_PLTBROWSER" = "xyes"])
 PRINT_PLUGIN_INFO([psf],[PSF player, using Audio Overload SDK],[test "x$HAVE_PSF" = "xyes"])
+PRINT_PLUGIN_INFO([pipewire],[Pipewire output plugin],[test "x$HAVE_PIPEWIRE_PLUGIN" = "xyes"])
 PRINT_PLUGIN_INFO([pulse],[PulseAudio output plugin],[test "x$HAVE_PULSEPLUGIN" = "xyes"])
 PRINT_PLUGIN_INFO([rgscanner],[plugin for ReplayGain scanner support],[test "x$HAVE_RGSCANNER" = "xyes"])
 PRINT_PLUGIN_INFO([sc68],[sc68 Atari ST And Amiga player],[test "x$HAVE_SC68" = "xyes"])
diff -rup '--exclude=.git' '--exclude=.gitmodules' deadbeef-orig/deadbeef.h deadbeef/deadbeef.h
--- deadbeef-orig/deadbeef.h	2023-06-14 11:01:10.134752040 +0300
+++ deadbeef/deadbeef.h	2023-06-14 10:43:51.340949831 +0300
@@ -782,6 +782,7 @@ typedef struct {
 
     // streamer access
     DB_playItem_t *(*streamer_get_playing_track) (void);
+    DB_playItem_t *(*streamer_get_playing_track_safe) (void);
     DB_playItem_t *(*streamer_get_streaming_track) (void);
     float (*streamer_get_playpos) (void);
     int (*streamer_ok_to_read) (int len);
diff -rup '--exclude=.git' '--exclude=.gitmodules' deadbeef-orig/external/Makefile.am deadbeef/external/Makefile.am
--- deadbeef-orig/external/Makefile.am	2023-06-14 11:01:10.134752040 +0300
+++ deadbeef/external/Makefile.am	2023-06-14 10:30:20.995556521 +0300
@@ -1,5 +1,13 @@
 noinst_LTLIBRARIES = libmp4p.la
 libmp4p_la_SOURCES = mp4p/src/mp4p.c mp4p/src/mp4patomdata.c mp4p/src/mp4pfile.c\
 				   	mp4p/include/mp4p/mp4p.h mp4p/include/mp4p/mp4patomdata.h mp4p/include/mp4p/mp4pfile.h 
-libmp4p_la_CFLAGS = -fPIC -std=c99 -I@top_srcdir@/external/mp4p/include
+libmp4p_la_CFLAGS = -fPIC -std=c99 -I@top_srcdir@/external/mp4p/include -I@top_srcdir@
 
+
+if HAVE_PIPEWIRE
+pkglib_LTLIBRARIES = ddb_out_pw.la
+ddb_out_pw_la_SOURCES = ddb_output_pw/pw.c
+ddb_out_pw_la_LDFLAGS = -module -avoid-version
+ddb_out_pw_la_CFLAGS = $(CFLAGS) $(PIPEWIRE_DEPS_CFLAGS) -std=c99
+ddb_out_pw_la_LIBADD = $(LDADD) $(PIPEWIRE_DEPS_LIBS)
+endif
diff -rup '--exclude=.git' '--exclude=.gitmodules' deadbeef-orig/plugins.c deadbeef/plugins.c
--- deadbeef-orig/plugins.c	2023-06-14 11:01:10.134752040 +0300
+++ deadbeef/plugins.c	2023-06-14 10:45:18.366255904 +0300
@@ -148,6 +148,7 @@ static DB_functions_t deadbeef_api = {
     .playback_set_pos = plug_playback_set_pos,
     // streamer access
     .streamer_get_playing_track = (DB_playItem_t *(*) (void))streamer_get_playing_track,
+    .streamer_get_playing_track_safe = (DB_playItem_t *(*) (void))streamer_get_playing_track,
     .streamer_get_streaming_track = (DB_playItem_t *(*) (void))streamer_get_streaming_track,
     .streamer_get_playpos = streamer_get_playpos,
     .streamer_ok_to_read = streamer_ok_to_read,
