diff -rupN deadbeef-1.8.8-orig/plugins.c deadbeef-1.8.8/plugins.c
--- deadbeef-1.8.8-orig/plugins.c	2023-02-20 09:44:40.476178773 +0200
+++ deadbeef-1.8.8/plugins.c	2023-02-20 09:44:52.084067441 +0200
@@ -1398,7 +1398,7 @@ _select_output_plugin (void) {
 #ifdef OSX_APPBUNDLE
     conf_get_str ("output_plugin", "coreaudio", outplugname, sizeof (outplugname));
 #else
-    conf_get_str ("output_plugin", "alsa", outplugname, sizeof (outplugname));
+    conf_get_str ("output_plugin", "pulse", outplugname, sizeof (outplugname));
 #endif
     for (int i = 0; g_output_plugins[i]; i++) {
         DB_output_t *p = g_output_plugins[i];
