diff -rupN uget-2.2.3-orig/uget-gtk.desktop uget-2.2.3/uget-gtk.desktop
--- uget-2.2.3-orig/uget-gtk.desktop	2023-01-17 20:15:06.969463252 +0200
+++ uget-2.2.3/uget-gtk.desktop	2023-01-17 20:15:24.550904369 +0200
@@ -2,7 +2,7 @@
 Name=uGet
 GenericName=Download Manager
 Comment=Download multiple URLs and apply it to one of setting/queue.
-Exec=env GDK_BACKEND=x11 uget-gtk %u
+Exec=uget-gtk %u
 Icon=uget-icon
 Terminal=false
 Type=Application
