diff -rupN xlockmore-5.66-orig/xlock/passwd.c xlockmore-5.66/xlock/passwd.c
--- xlockmore-5.66-orig/xlock/passwd.c	2022-03-06 13:27:50.906407728 +0200
+++ xlockmore-5.66/xlock/passwd.c	2022-03-06 14:16:34.069494306 +0200
@@ -55,6 +55,7 @@ extern Bool inwindow;
 extern Bool grabmouse;
 extern Bool nolock;
 extern char *cpasswd;
+extern char *startCmd;
 
 #ifdef USE_PAM
 #include <X11/Xlib.h>
@@ -731,7 +732,19 @@ gpass(void)
 		(void) strncpy(xlockrc, home, MAXPATHLEN);
 		xlockrc[MAXPATHLEN] = '\0';
 		(void) strncat(xlockrc, "/.xlockrc", MAXPATHLEN);
-		if ((fp = my_fopen(xlockrc, "r")) == NULL) {
+		fp = my_fopen(xlockrc, "r");
+		if (fp != NULL) {
+			if ((fseek(fp, 0, SEEK_END) != 0) || (ftell(fp) == 0)) {
+				fclose(fp);
+				fp = NULL;
+			} else {
+				rewind(fp);
+			}
+		}
+		if (fp == NULL) {
+			if (startCmd && startCmd[0])
+				return;
+
 			if ((fp = my_fopen(xlockrc, "w")) != NULL)
 				(void) fchmod(fileno(fp), 0600);
 #if defined(HAVE_KRB4) || defined(HAVE_KRB5)
@@ -739,7 +752,7 @@ gpass(void)
 #else
 			if (!gpasskey(userpass)) {
 				remove(xlockrc); /* else creates annoying null file */
-				exit(1);
+				goto fail;
 			}
 #endif /* KRB4 || KRB5 */
 			if (fp)
@@ -761,15 +774,16 @@ gpass(void)
 #if defined(HAVE_KRB4) || defined(HAVE_KRB5)
 				if (!gpasskey(buf)) {
 					remove(xlockrc); /* else creates annoying null file */
-					exit(1);
+					goto fail;
 				}
 #else
-				exit(1);
+				goto fail;
 #endif
 			}
 			buf[CPASSLENGTH - 1] = '\0';
 			(void) strncpy(userpass, buf, CPASSLENGTH);
 		}
+fail:
 		if (fp)
 			(void) fclose(fp);
 	} else {
