diff -rupN activate-linux-2e8716dc7ba9a39bd0aaff56918bfe8e7c1ad892-orig/src/activate_linux.c activate-linux-2e8716dc7ba9a39bd0aaff56918bfe8e7c1ad892/src/activate_linux.c
--- activate-linux-2e8716dc7ba9a39bd0aaff56918bfe8e7c1ad892-orig/src/activate_linux.c	2022-08-15 12:55:52.724311523 +0000
+++ activate-linux-2e8716dc7ba9a39bd0aaff56918bfe8e7c1ad892/src/activate_linux.c	2022-08-15 12:57:33.344479457 +0000
@@ -15,8 +15,8 @@ void draw_window(GdkMonitor *monitor, ch
 	gtk_layer_set_layer(gtk_window, GTK_LAYER_SHELL_LAYER_TOP);
 
 	// gtk layer shell options
-	gtk_layer_set_margin(gtk_window, GTK_LAYER_SHELL_EDGE_RIGHT, 40);
-	gtk_layer_set_margin(gtk_window, GTK_LAYER_SHELL_EDGE_BOTTOM, 40);
+	gtk_layer_set_margin(gtk_window, GTK_LAYER_SHELL_EDGE_RIGHT, 10);
+	gtk_layer_set_margin(gtk_window, GTK_LAYER_SHELL_EDGE_BOTTOM, 10);
 	static const gboolean anchors[] = {FALSE, TRUE, FALSE, TRUE};
 	for (int i = 0; i < GTK_LAYER_SHELL_EDGE_ENTRY_NUMBER; i++) {
 		gtk_layer_set_anchor(gtk_window, i, anchors[i]);
@@ -34,6 +34,7 @@ void draw_window(GdkMonitor *monitor, ch
 
 	// render label
 	GtkWidget *label = gtk_label_new("");
+	gtk_label_set_justify(GTK_LABEL(label), GTK_JUSTIFY_RIGHT);
 	gtk_label_set_markup(GTK_LABEL(label), text);
 	gtk_container_add(GTK_CONTAINER(gtk_window), label);
 	gtk_container_set_border_width(GTK_CONTAINER(gtk_window), 12);
@@ -49,7 +50,7 @@ static void activate(GtkApplication *app
 	int len = 100+title_len+subtitle_len;
 	char text[len];
 
-	snprintf(text, len, "<span alpha=\"50%\" font_desc=\"24.0\">%s</span>\n<span alpha=\"50%\" font_desc=\"16\">%s</span>", conf->title, conf->subtitle);
+	snprintf(text, len, "<span alpha=\"50%\" font_desc=\"8.0\">%s</span>\n<span alpha=\"50%\" font_desc=\"8\">%s</span>", conf->title, conf->subtitle);
 
 	GdkDisplay *display = gdk_display_get_default();
 	for (int i = 0; i < gdk_display_get_n_monitors(display); i++) {
