diff -Naur ntfs-3g_ntfsprogs-2016.2.22/libfuse-lite/fusermount.c ntfs-3g_ntfsprogs-2016.2.22-patched/libfuse-lite/fusermount.c
--- ntfs-3g_ntfsprogs-2016.2.22/libfuse-lite/fusermount.c	2016-02-22 07:34:33.000000000 +0000
+++ ntfs-3g_ntfsprogs-2016.2.22-patched/libfuse-lite/fusermount.c	2016-04-18 11:02:51.000000000 +0000
@@ -36,6 +36,10 @@
 
 #define FUSE_DEV_NEW "/dev/fuse"
 
+#ifndef _PATH_MOUNTED
+#define _PATH_MOUNTED "/proc/mounts"
+#endif
+
 #ifndef MS_DIRSYNC
 #define MS_DIRSYNC 128
 #endif
diff -Naur ntfs-3g_ntfsprogs-2016.2.22/libfuse-lite/mount_util.c ntfs-3g_ntfsprogs-2016.2.22-patched/libfuse-lite/mount_util.c
--- ntfs-3g_ntfsprogs-2016.2.22/libfuse-lite/mount_util.c	2016-02-22 07:34:33.000000000 +0000
+++ ntfs-3g_ntfsprogs-2016.2.22-patched/libfuse-lite/mount_util.c	2016-04-18 11:02:51.000000000 +0000
@@ -263,6 +263,10 @@
 
 #else /* __SOLARIS__ */
 
+#ifndef _PATH_MOUNTED
+#define _PATH_MOUNTED "/proc/mounts"
+#endif /* _PATH_MOUNTED */
+
 static int mtab_needs_update(const char *mnt)
 {
 	int res;
