From 5a2f47eea6f4bc61c19fd1d06e5cf171d8ab1018 Mon Sep 17 00:00:00 2001
From: Dima Krasner <dima@dimakrasner.com>
Date: Fri, 1 Apr 2022 20:48:53 +0800
Subject: [PATCH] ask clients to draw their own decorations

---
 dwl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dwl.c b/dwl.c
index ac62120..52b4d27 100644
--- a/dwl.c
+++ b/dwl.c
@@ -2003,7 +2003,7 @@ setup(void)
 	/* Use decoration protocols to negotiate server-side decorations */
 	wlr_server_decoration_manager_set_default_mode(
 			wlr_server_decoration_manager_create(dpy),
-			WLR_SERVER_DECORATION_MANAGER_MODE_SERVER);
+			WLR_SERVER_DECORATION_MANAGER_MODE_CLIENT);
 	wlr_xdg_decoration_manager_v1_create(dpy);
 
 	/*
