--- 3builddistro-Z	2017-03-20 19:26:32.308459487 -0500
+++ ../../woof-out_x86_64_arm_raspbian_stretch/3builddistro-Z	2017-03-21 23:23:15.012888807 -0500
@@ -1715,6 +1715,7 @@
 
 sync
 
+if [ "$WOOF_TARGETARCH" != 'arm' ] ; then
 # Debian builds have executables as shared libraries. ROX gets confused
 if [ "$DISTRO_COMPAT_VERSION" = "stretch" -o "$DISTRO_COMPAT_VERSION" = "ascii" -o "$DISTRO_COMPAT_VERSION" = "xenial" ]; then
 	if [ "$(which elfedit)" ];then
@@ -1734,6 +1735,7 @@
 		 "$(gettext 'Please install elfedit and try later.')"
 	fi
 fi
+fi
 
 
 if [ "$BUILD_SFS" = 'yes' ]; then
@@ -2228,6 +2230,7 @@
 echo -e "\ncleaning out whiteouts"
 find sandbox3/devx -name '.wh*' -delete
 
+if [ "$WOOF_TARGETARCH" != 'arm' ] ; then
 # Debian builds have executables as shared libraries. ROX gets confused
 if [ "$DISTRO_COMPAT_VERSION" = "stretch" -o "$DISTRO_COMPAT_VERSION" = "ascii" -o "$DISTRO_COMPAT_VERSION" = "xenial" ]; then
 	if [ "$(which elfedit)" ];then
@@ -2247,6 +2250,7 @@
 		 "$(gettext 'Please install elfedit and try later.')"
 	fi
 fi
+fi
 
 echo "Now creating ${DEVXSFS} ..."
 ${MKSQUASHFS} sandbox3/devx ./${WOOF_OUTPUT}/${DEVXSFS} ${SFSCOMP} #100911 110713
