--- 3builddistro-Z	2017-03-20 19:26:32.308459487 -0500
+++ ../../woof-out_x86_64_arm_raspbian_stretch/3builddistro-Z	2017-03-20 18:50:00.781010718 -0500
@@ -2148,7 +2148,14 @@
 #think should delete them if they really point nowhere...
 echo " Deleting really invalid symlinks in devx..."
 mkdir layer_top
-mount -t aufs -o udba=reval,diropq=w,dirs=sandbox3/devx=rw:sandbox3/rootfs-complete=ro layerfs layer_top
+
+if [ "$LAYER_TYPE" = 'overlay' ]; then
+	mkdir overlay_workdir
+	mount -t overlay overlay -olowerdir=sandbox3/rootfs-complete,upperdir=sandbox3/devx,workdir=overlay_workdir layer_top
+else
+	mount -t aufs -o udba=reval,diropq=w,dirs=sandbox3/devx=rw:sandbox3/rootfs-complete=ro layerfs layer_top
+fi
+
 if [ "$WOOF_HOSTARCH" = "$WOOF_TARGETARCH" ] ; then
 	dirs=$(ls -d layer_top/* | sed 's|layer_top||' | grep -vE '/dev/|/proc/|/sys/') #|/initrd/|/tmp/|/var/|/run/|/mnt/
 	chroot layer_top find -L $dirs -type l -delete
@@ -2226,7 +2233,20 @@
 done
 sync
 echo -e "\ncleaning out whiteouts"
+# aufs whiteouts
 find sandbox3/devx -name '.wh*' -delete
+# overlayfs whiteouts
+if [ "$LAYER_TYPE" = 'overlay' ]; then
+	for ONE_CHAR_NODE in `find sandbox3/devx -type c`
+	do
+		if [ "`stat -c '%t, %T' $ONE_CHAR_NODE`" = '0, 0' ]; then
+			# paranoid test, regular files are also '0, 0'
+			if [ "`stat -c '%F' $ONE_CHAR_NODE`" = 'character special file' ]; then
+				rm $ONE_CHAR_NODE
+			fi
+		fi
+	done
+fi
 
 # Debian builds have executables as shared libraries. ROX gets confused
 if [ "$DISTRO_COMPAT_VERSION" = "stretch" -o "$DISTRO_COMPAT_VERSION" = "ascii" -o "$DISTRO_COMPAT_VERSION" = "xenial" ]; then
