--- initrd-progs/0initrd/init	2018-02-19 08:20:48.079752378 -0600
+++ ../fake/woof-out_x86_64_arm_raspbian_stretch/initrd-progs/0initrd/init	2018-02-27 22:31:02.291561671 -0600
@@ -68,6 +68,7 @@
 # it's added by puppyinstaller (or you can add it manually)
 for i in $(cat /proc/cmdline) ; do
   [ "$i" = "fullinstall" ] && exec /init_full_install
+  [ "$i" = "bootmenu" -a -z "$BOOTMENU_DONE" ] && exec /init-bootmenu
 done
 
 export TERM="xterm"
