--- initrd-progs/0initrd/init	2019-03-09 15:39:22.584879363 -0600
+++ ../fake/woof-out_arm_arm_raspbian_stretch/initrd-progs/0initrd/init	2019-03-09 18:45:16.517504388 -0600
@@ -63,7 +63,10 @@
 # it's added by puppyinstaller (or you can add it manually)
 for i in $(cat /proc/cmdline) ; do
   [ "$i" = "fullinstall" ] && exec /init_full_install
+  [ "$i" = "bootmenu" -a -z "$BOOTMENU_DONE" ] && exec /init-bootmenu
 done
+# always call init-bootmenu on ARM devices
+[ "$(uname -m | grep arm)" != "" -a -z "$BOOTMENU_DONE" ] && exec /init-bootmenu
 
 export TERM="xterm"
 export TERMINFO="/usr/share/terminfo"
