--- rootfs-skeleton/etc/rc.d/rc.sysinit	2016-12-21 14:23:55.000000000 -0600
+++ ../../woof-out_x86_arm_raspbian_jessie/rootfs-skeleton/etc/rc.d/rc.sysinit	2016-12-25 15:53:49.000000000 -0600
@@ -284,7 +284,7 @@
 
 #Ensure kernel-specific firmware.dep file present.
 [ ! -e /etc/modules/firmware.dep.$KERNVER -a -f /etc/modules/firmware.dep ] \
- && mv -fv /etc/modules/firmware.dep /etc/modules/firmware.dep.$KERNVER
+ && cp -fv /etc/modules/firmware.dep /etc/modules/firmware.dep.$KERNVER
 
 #################LOAD KERNEL MODULES################
 echo "LOAD KERNEL MODULES"
