--- 0setup	2017-01-04 22:00:30.000000000 -0600
+++ ../../woof-out_x86_arm_raspbian_stretch/0setup	2017-01-04 23:14:01.000000000 -0600
@@ -94,6 +94,7 @@
 #each line: pkgname|nameonly|version|pkgrelease|category[;subcategory]|size|path|fullfilename|dependencies|description|
 #ex: abiword-1.2.4|abiword|1.2.4|5|Document|999K|slackware/ab|abiword-1.2.4-5-i486.tgz|+aiksausus,+gtk2|a nice wordprocessor|
 PKGLISTS_COMPAT="`echo "$PKG_DOCS_DISTRO_COMPAT" | tr ' ' '\n' | cut -f 3 -d '|' | tr '\n' ' '`" #see file DISTRO_PKGS_SPECS-ubuntu
+[ -f "Packages-${DISTRO_BINARY_COMPAT}-${DISTRO_COMPAT_VERSION}-overrides" ] && PKGLISTS_COMPAT="Packages-${DISTRO_BINARY_COMPAT}-${DISTRO_COMPAT_VERSION}-overrides $PKGLISTS_COMPAT"
 
 function check_compat_distro_pkgs() {
 	####################################

--- support/findpkgs	2017-01-04 22:00:31.000000000 -0600
+++ ../../woof-out_x86_arm_raspbian_stretch/support/findpkgs	2017-01-04 23:20:24.000000000 -0600
@@ -94,6 +94,7 @@
 #the files that have compatible-distro pkg docs (these were downloaded by 0setup)...
 PKGLISTS_COMPAT="`echo "$PKG_DOCS_DISTRO_COMPAT" | tr ' ' '\n' | cut -f 3 -d '|' | tr '\n' ' '`" #see file DISTRO_PKGS_SPECS-ubuntu
 #...format ex: 'Packages-ubuntu-intrepid-main Packages-ubuntu-intrepid-universe'
+[ -f "Packages-${DISTRO_BINARY_COMPAT}-${DISTRO_COMPAT_VERSION}-overrides" ] && PKGLISTS_COMPAT="Packages-${DISTRO_BINARY_COMPAT}-${DISTRO_COMPAT_VERSION}-overrides $PKGLISTS_COMPAT"
 
 CURRDIR="`pwd`"
 mkdir -p status
