--- rootfs-skeleton/etc/rc.d/rc.sysinit	2017-03-17 18:50:51.788098486 -0500
+++ ../../woof-out_x86_64_arm_raspbian_stretch/rootfs-skeleton/etc/rc.d/rc.sysinit	2017-03-29 17:34:01.463548649 -0500
@@ -498,6 +498,7 @@
   echo "Warning: Xorg kbd_drv.so and mouse_drv.so are missing"
   XORG_UDEV=1
 fi
+[ "$DISTRO_BINARY_COMPAT" = 'raspbian' ] && XORG_UDEV=1
 if [ "$XORG_UDEV" != "" ] ; then
   echo "Activating Xorg Evdev"
   [ ! -f /var/local/xorg_udev ] && rm -fv /etc/X11/xorg.conf #force xorgwizard-automatic
