diff -rupN gcolor3-00e737f2273aa0dcb04bf75602e8f44ef6237626-orig/meson.build gcolor3-00e737f2273aa0dcb04bf75602e8f44ef6237626/meson.build
--- gcolor3-00e737f2273aa0dcb04bf75602e8f44ef6237626-orig/meson.build	2023-02-19 19:35:32.373273908 +0200
+++ gcolor3-00e737f2273aa0dcb04bf75602e8f44ef6237626/meson.build	2023-02-19 19:36:33.095694830 +0200
@@ -4,7 +4,6 @@ project('gcolor3', 'c', version: '2.4.0'
 )
 
 dep_gtk = dependency('gtk+-3.0', version: '>= 3.20.0', required: true)
-dep_handy = dependency('libhandy-1', version: '>= 1.5')
 dep_libportal = dependency(
   'libportal',
   required: true,
diff -rupN gcolor3-00e737f2273aa0dcb04bf75602e8f44ef6237626-orig/src/gcolor3-application.c gcolor3-00e737f2273aa0dcb04bf75602e8f44ef6237626/src/gcolor3-application.c
--- gcolor3-00e737f2273aa0dcb04bf75602e8f44ef6237626-orig/src/gcolor3-application.c	2023-02-19 19:35:32.373273908 +0200
+++ gcolor3-00e737f2273aa0dcb04bf75602e8f44ef6237626/src/gcolor3-application.c	2023-02-19 19:36:54.396334606 +0200
@@ -23,7 +23,6 @@
 
 #include <gtk/gtk.h>
 #include <glib/gi18n.h>
-#include <libhandy-1/handy.h>
 
 #include "gcolor3-application.h"
 #include "gcolor3-color-store.h"
@@ -146,10 +145,6 @@ gcolor3_application_startup (GApplicatio
 
 	G_APPLICATION_CLASS (gcolor3_application_parent_class)->startup (application);
 
-	hdy_init ();
-	hdy_style_manager_set_color_scheme (hdy_style_manager_get_default (),
-										HDY_COLOR_SCHEME_PREFER_LIGHT);
-
 	gtk_window_set_default_icon_name ("gcolor3");
 	g_set_application_name (_("Color Picker"));
 
diff -rupN gcolor3-00e737f2273aa0dcb04bf75602e8f44ef6237626-orig/src/meson.build gcolor3-00e737f2273aa0dcb04bf75602e8f44ef6237626/src/meson.build
--- gcolor3-00e737f2273aa0dcb04bf75602e8f44ef6237626-orig/src/meson.build	2023-02-19 19:35:32.373273908 +0200
+++ gcolor3-00e737f2273aa0dcb04bf75602e8f44ef6237626/src/meson.build	2023-02-19 19:36:38.014918811 +0200
@@ -41,7 +41,6 @@ executable(
   ],
   dependencies: [
     dep_gtk,
-    dep_handy,
     dep_libportal,
     dep_libportal_gtk3,
     dep_lm
