diff -rupN weechat-3.5-orig/src/plugins/irc/irc-ctcp.c weechat-3.5/src/plugins/irc/irc-ctcp.c
--- weechat-3.5-orig/src/plugins/irc/irc-ctcp.c	2022-07-06 08:12:57.938081298 +0300
+++ weechat-3.5/src/plugins/irc/irc-ctcp.c	2022-07-06 08:16:15.954853936 +0300
@@ -40,12 +40,12 @@
 
 
 struct t_irc_ctcp_reply irc_ctcp_default_reply[] =
-{ { "clientinfo", "$clientinfo" },
-  { "finger",     "WeeChat $versiongit" },
-  { "source",     "$download" },
-  { "time",       "$time" },
-  { "userinfo",   "$username ($realname)" },
-  { "version",    "WeeChat $versiongit ($compilation)" },
+{ { "clientinfo", "" },
+  { "finger",     "" },
+  { "source",     "" },
+  { "time",       "" },
+  { "userinfo",   "" },
+  { "version",    "" },
   { NULL,         NULL },
 };
 
diff -rupN weechat-3.5-orig/src/plugins/irc/irc-server.c weechat-3.5/src/plugins/irc/irc-server.c
--- weechat-3.5-orig/src/plugins/irc/irc-server.c	2022-07-06 08:12:57.938081298 +0300
+++ weechat-3.5/src/plugins/irc/irc-server.c	2022-07-06 08:15:50.331862125 +0300
@@ -116,8 +116,8 @@ char *irc_server_options[IRC_SERVER_NUM_
   { "away_check",           "0"                       },
   { "away_check_max_nicks", "25"                      },
   { "msg_kick",             ""                        },
-  { "msg_part",             "WeeChat ${info:version}" },
-  { "msg_quit",             "WeeChat ${info:version}" },
+  { "msg_part",             ""                        },
+  { "msg_quit",             ""                        },
   { "notify",               ""                        },
   { "split_msg_max_length", "512"                     },
   { "charset_message",      "message"                 },
