diff -rupN labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2-orig/meson.build labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2/meson.build
--- labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2-orig/meson.build	2021-11-08 18:12:16.516047739 +0200
+++ labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2/meson.build	2021-11-08 18:12:43.972046853 +0200
@@ -38,22 +38,9 @@ endif
 add_project_arguments('-DLABWC_VERSION=@0@'.format(version), language: 'c')
 
 wlroots_version = ['>=0.15.0', '<0.16.0']
-wlroots_proj = subproject(
-  'wlroots',
-  default_options: ['examples=false'],
-  required: false,
-  version: wlroots_version,
-)
-
-if wlroots_proj.found()
-  wlroots = wlroots_proj.get_variable('wlroots')
-  wlroots_conf = wlroots_proj.get_variable('conf_data')
-  wlroots_has_xwayland = wlroots_conf.get('WLR_HAS_XWAYLAND') == 1
-else
-  wlroots = dependency('wlroots', version: wlroots_version)
-  wlroots_has_xwayland = cc.get_define('WLR_HAS_XWAYLAND', prefix: '#include <wlr/config.h>', dependencies: wlroots) == '1'
-endif
-wayland_server = dependency('wayland-server', version: '>=1.19.0')
+wlroots = dependency('wlroots', version: wlroots_version)
+wlroots_has_xwayland = cc.get_define('WLR_HAS_XWAYLAND', prefix: '#include <wlr/config.h>', dependencies: wlroots) == '1'
+wayland_server = dependency('wayland-server', version: '>=1.18.0')
 wayland_protos = dependency('wayland-protocols')
 xkbcommon = dependency('xkbcommon')
 xcb = dependency('xcb', required: get_option('xwayland'))
