diff -rupN labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2-orig/src/server.c labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2/src/server.c
--- labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2-orig/src/server.c	2021-11-08 17:55:59.328079274 +0200
+++ labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2/src/server.c	2021-11-08 17:56:30.376078272 +0200
@@ -66,6 +66,8 @@ handle_sigterm(int signal, void *data)
 static void
 drop_permissions(void)
 {
+#if 0
+
 	if (getuid() != geteuid() || getgid() != getegid()) {
 		if (setgid(getgid())) {
 			wlr_log(WLR_ERROR, "unable to drop root group");
@@ -80,6 +82,7 @@ drop_permissions(void)
 		wlr_log(WLR_ERROR, "unable to drop root");
 		exit(EXIT_FAILURE);
 	}
+#endif
 }
 
 static void
