diff -rupN labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2-orig/include/labwc.h labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2/include/labwc.h
--- labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2-orig/include/labwc.h	2021-11-08 20:03:12.721870261 +0200
+++ labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2/include/labwc.h	2021-11-08 20:05:17.913873237 +0200
@@ -342,7 +342,7 @@ void xwayland_unmanaged_create(struct se
 void view_child_init(struct view_child *child, struct view *view,
 	struct wlr_surface *wlr_surface);
 void view_child_finish(struct view_child *child);
-void subsurface_create(struct view *view, struct wlr_subsurface *wlr_subsurface);
+void view_subsurface_create(struct view *view, struct wlr_subsurface *wlr_subsurface);
 
 void view_set_activated(struct view *view, bool activated);
 struct border view_border(struct view *view);
diff -rupN labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2-orig/src/subsurface.c labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2/src/subsurface.c
--- labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2-orig/src/subsurface.c	2021-11-08 20:03:12.725870261 +0200
+++ labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2/src/subsurface.c	2021-11-08 20:05:35.285873650 +0200
@@ -21,7 +21,7 @@ subsurface_handle_destroy(struct wl_list
 }
 
 void
-subsurface_create(struct view *view, struct wlr_subsurface *wlr_subsurface)
+view_subsurface_create(struct view *view, struct wlr_subsurface *wlr_subsurface)
 {
 	struct view_subsurface *subsurface =
 		calloc(1, sizeof(struct view_subsurface));
diff -rupN labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2-orig/src/view-child.c labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2/src/view-child.c
--- labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2-orig/src/view-child.c	2021-11-08 20:03:12.725870261 +0200
+++ labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2/src/view-child.c	2021-11-08 20:05:00.913872833 +0200
@@ -19,7 +19,7 @@ view_child_handle_new_subsurface(struct
 	struct view_child *child;
 	child = wl_container_of(listener, child, new_subsurface);
 	struct wlr_subsurface *wlr_subsurface = data;
-	subsurface_create(child->parent, wlr_subsurface);
+	view_subsurface_create(child->parent, wlr_subsurface);
 }
 
 void
diff -rupN labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2-orig/src/xdg.c labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2/src/xdg.c
--- labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2-orig/src/xdg.c	2021-11-08 20:03:12.725870261 +0200
+++ labwc-b878db57a7c84ff8daf6a442caca013615aa5bd2/src/xdg.c	2021-11-08 20:04:54.173872673 +0200
@@ -4,7 +4,7 @@
 #include "ssd.h"
 
 /*
- * xdg_popup_create() and subsurface_create() are only called for the
+ * xdg_popup_create() and view_subsurface_create() are only called for the
  * purposes of tracking damage.
  */
 static void
@@ -20,7 +20,7 @@ new_subsurface_notify(struct wl_listener
 {
 	struct view *view = wl_container_of(listener, view, new_subsurface);
 	struct wlr_subsurface *wlr_subsurface = data;
-	subsurface_create(view, wlr_subsurface);
+	view_subsurface_create(view, wlr_subsurface);
 }
 
 static bool
@@ -337,11 +337,11 @@ xdg_toplevel_view_map(struct view *view)
 		struct wlr_subsurface *subsurface;
 		wl_list_for_each(subsurface, &view->surface->current.subsurfaces_below,
 				 current.link) {
-			subsurface_create(view, subsurface);
+			view_subsurface_create(view, subsurface);
 		}
 		wl_list_for_each(subsurface, &view->surface->current.subsurfaces_above,
 				 current.link) {
-			subsurface_create(view, subsurface);
+			view_subsurface_create(view, subsurface);
 		}
 
 		view->been_mapped = true;
