diff -rup xpad-4.3.1-orig/src/xpad-app.c xpad-4.3.1/src/xpad-app.c
--- xpad-4.3.1-orig/src/xpad-app.c	2024-08-12 20:52:34.180617225 +0300
+++ xpad-4.3.1/src/xpad-app.c	2024-08-12 20:53:18.864466842 +0300
@@ -718,7 +718,7 @@ xpad_app_pass_args (void)
 	guint size;
 	ssize_t bytesRead;
 	gboolean connected = FALSE;
-	ssize_t error = NULL;
+	ssize_t error = 0;
 	
 	/* create master socket */
 	client_fd = socket (PF_LOCAL, SOCK_STREAM, 0);
diff -rup xpad-4.3.1-orig/src/xpad-pad.c xpad-4.3.1/src/xpad-pad.c
--- xpad-4.3.1-orig/src/xpad-pad.c	2024-08-12 20:52:34.180617225 +0300
+++ xpad-4.3.1/src/xpad-pad.c	2024-08-12 20:53:55.764357507 +0300
@@ -1347,7 +1347,7 @@ void
 xpad_pad_save_info (XpadPad *pad)
 {
 	gboolean follow_font_style, follow_color_style;
-	guint height = NULL;
+	guint height = 0;
 	GtkStyleContext *style = NULL;
 	PangoFontDescription *font = NULL;
 	GdkRGBA text_color = {0, 0, 0, 0}, back_color = {0, 0, 0, 0};
