diff -rupN xpad-4.3.1-orig/src/xpad-app.c xpad-4.3.1/src/xpad-app.c
--- xpad-4.3.1-orig/src/xpad-app.c	2023-01-10 08:16:25.193507123 +0200
+++ xpad-4.3.1/src/xpad-app.c	2023-01-10 08:17:06.881477512 +0200
@@ -107,7 +107,7 @@ xpad_app_init (int argc, char **argv)
 	output = stdout;
 	
 	/* Set up config directory. */
-	first_time = !config_dir_exists ();
+	first_time = FALSE;
 	config_dir = make_config_dir ();
 	
 	/* create master socket name */
