diff -rupN xpad-4.3.1-orig/src/xpad-session-manager.c xpad-4.3.1/src/xpad-session-manager.c
--- xpad-4.3.1-orig/src/xpad-session-manager.c	2023-01-07 12:26:19.714891083 +0200
+++ xpad-4.3.1/src/xpad-session-manager.c	2023-01-07 12:26:40.674871463 +0200
@@ -19,10 +19,11 @@ Foundation, Inc., 59 Temple Place, Suite
 
 */
 
-#ifndef X_DISPLAY_MISSING
-
 #include "../config.h"
 #include "xpad-session-manager.h"
+
+#if !X_DISPLAY_MISSING
+
 #include <pwd.h>
 #include <stdlib.h>
 #include <string.h>
