diff -rupN connman-ui-fce0af94e121bde77c7fa2ebd6a319f0180c5516-orig/src/left-menu.c connman-ui-fce0af94e121bde77c7fa2ebd6a319f0180c5516/src/left-menu.c
--- connman-ui-fce0af94e121bde77c7fa2ebd6a319f0180c5516-orig/src/left-menu.c	2021-09-24 08:41:00.046259198 +0300
+++ connman-ui-fce0af94e121bde77c7fa2ebd6a319f0180c5516/src/left-menu.c	2021-09-24 08:42:01.778185449 +0300
@@ -235,7 +235,7 @@ static void cui_popup_left_menu(GtkStatu
 							scanning_cb, user_data);
 
 	gtk_menu_popup(cui_left_menu, NULL, NULL,
-			menu_position_func, trayicon, 1, 0);
+			menu_position_func, trayicon, 1, gtk_get_current_event_time());
 }
 
 static void cui_popdown_left_menu(GtkMenu *menu, gpointer user_data)
