diff -rupN connman-ui-fce0af94e121bde77c7fa2ebd6a319f0180c5516-orig/src/left-menu.c connman-ui-fce0af94e121bde77c7fa2ebd6a319f0180c5516/src/left-menu.c
--- connman-ui-fce0af94e121bde77c7fa2ebd6a319f0180c5516-orig/src/left-menu.c	2021-09-25 17:54:53.484028459 +0800
+++ connman-ui-fce0af94e121bde77c7fa2ebd6a319f0180c5516/src/left-menu.c	2021-09-25 17:55:09.624028843 +0800
@@ -28,6 +28,7 @@ static GtkMenu *cui_left_menu = NULL;
 static GtkMenu *cui_more_menu = NULL;
 static GtkMenuItem *cui_list_more_item = NULL;
 static GHashTable *service_items = NULL;
+static GtkMenuItem *cui_service_list = NULL;
 static GtkMenuItem *cui_scan_spinner = NULL;
 
 static void add_or_update_service(const char *path, int position)
@@ -204,6 +205,7 @@ static void scanning_cb(void *user_data)
 	gtk_spinner_stop(spin);
 	gtk_widget_hide((GtkWidget *)cui_scan_spinner);
 	gtk_widget_hide((GtkWidget *)spin);
+	gtk_widget_hide((GtkWidget *)cui_service_list);
 
 	/* Reposition left menu after hidding the spinner */
 	gtk_menu_reposition(cui_left_menu);
@@ -226,6 +228,7 @@ static void cui_popup_left_menu(GtkStatu
 	gtk_widget_hide(GTK_WIDGET(cui_list_more_item));
 	gtk_widget_show((GtkWidget *)cui_scan_spinner);
 	gtk_widget_show((GtkWidget *)spin);
+	gtk_widget_show((GtkWidget *)cui_service_list);
 
 	gtk_spinner_start(spin);
 
@@ -263,6 +266,8 @@ gint cui_load_left_menu(GtkBuilder *buil
 							"cui_more_menu");
 	cui_list_more_item = (GtkMenuItem *) gtk_builder_get_object(builder,
 							"cui_list_more_item");
+	cui_service_list = (GtkMenuItem *) gtk_builder_get_object(builder,
+							"service_list_menu_item");
 	cui_scan_spinner = (GtkMenuItem *) gtk_builder_get_object(builder,
 							"cui_scan_spinner");
 
