diff -rup gtkdialog-0.8.4a-orig/Makefile.in gtkdialog-0.8.4a/Makefile.in
--- gtkdialog-0.8.4a-orig/Makefile.in	2020-12-26 11:48:42.104315202 +0200
+++ gtkdialog-0.8.4a/Makefile.in	2020-12-26 11:48:51.828231585 +0200
@@ -309,7 +309,7 @@ target_alias = @target_alias@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = src doc data
+SUBDIRS = src data
 EXTRA_DIST = \
 	autogen.sh \
 	install.iss pango.modules pango.aliases gdk-pixbuf.loaders \
