diff -rup chromiumos_kernel-chromeos-3.14-git161015-orig/kernel/printk/printk.c chromiumos_kernel-chromeos-3.14-git161015/kernel/printk/printk.c
--- chromiumos_kernel-chromeos-3.14-git161015-orig/kernel/printk/printk.c	2015-10-16 20:45:59.630980612 +0300
+++ chromiumos_kernel-chromeos-3.14-git161015/kernel/printk/printk.c	2015-10-16 20:48:03.847596272 +0300
@@ -64,7 +64,7 @@ extern void printascii(char *);
 
 /* We show everything that is MORE important than this.. */
 #define MINIMUM_CONSOLE_LOGLEVEL 1 /* Minimum loglevel we let people use */
-#define DEFAULT_CONSOLE_LOGLEVEL 7 /* anything MORE serious than KERN_DEBUG */
+#define DEFAULT_CONSOLE_LOGLEVEL 3 /* anything MORE serious than KERN_ERR */
 
 int console_printk[4] = {
 	DEFAULT_CONSOLE_LOGLEVEL,	/* console_loglevel */
