--- 3builddistro-Z	2016-12-12 15:29:12.000000000 -0600
+++ ../../woof-out_x86_arm_raspbian_jessie/3builddistro-Z	2016-12-12 16:45:02.000000000 -0600
@@ -1383,7 +1383,7 @@
 echo "Removing redundant firmware as it is in the z drive"
 rm -rf rootfs-complete/lib/firmware 2>/dev/null
 #150624 remove redundant stuff from 'all-firmware'
-for fw in agrsm hso iwlwifi libertas* madwifi* mwl8k mwave* nozomi RTL8192E rt2x00 rtl_nic rtlwifi brcm;do
+for fw in agrsm hso iwlwifi libertas* madwifi* mwl8k mwave* nozomi RTL8192E rt2x00 rtl_nic rtlwifi;do
 	echo "removing $fw"
 	rm -rf rootfs-complete/lib/modules/all-firmware/${fw} 2>/dev/null
 done
