diff -rupN gnome-menus-2.14.3-orig/libmenu/desktop-entries.c gnome-menus-2.14.3/libmenu/desktop-entries.c
--- gnome-menus-2.14.3-orig/libmenu/desktop-entries.c	2005-12-08 10:39:41.000000000 +0200
+++ gnome-menus-2.14.3/libmenu/desktop-entries.c	2021-10-01 18:52:37.024744848 +0300
@@ -51,6 +51,8 @@ struct DesktopEntry
   guint type : 2;
   guint flags : 4;
   guint refcount : 24;
+
+  gboolean terminal;
 };
 
 struct DesktopEntrySet
@@ -278,7 +280,10 @@ desktop_entry_load (DesktopEntry *entry)
   retval->categories = get_categories_from_key_file (retval, key_file, desktop_entry_group);
 
   if (entry->type == DESKTOP_ENTRY_DESKTOP)
-    retval->exec = g_key_file_get_string (key_file, desktop_entry_group, "Exec", NULL);
+    {
+      retval->exec = g_key_file_get_string (key_file, desktop_entry_group, "Exec", NULL);
+      retval->terminal = g_key_file_get_boolean (key_file, desktop_entry_group, "Terminal", NULL);
+    }
   
 #undef GET_LOCALE_STRING
 
@@ -356,6 +361,7 @@ desktop_entry_reload (DesktopEntry *entr
   g_free (entry->exec);
   entry->exec = NULL;
 
+  entry->terminal = 0;
   entry->flags = 0;
 
   return desktop_entry_load (entry);
@@ -391,6 +397,7 @@ desktop_entry_copy (DesktopEntry *entry)
   retval->comment  = g_strdup (entry->comment);
   retval->icon     = g_strdup (entry->icon);
   retval->exec     = g_strdup (entry->exec);
+  retval->terminal = entry->terminal;
   retval->flags    = entry->flags;
 
   i = 0;
@@ -488,6 +495,12 @@ desktop_entry_get_exec (DesktopEntry *en
 }
 
 gboolean
+desktop_entry_get_launch_in_terminal (DesktopEntry *entry)
+{
+  return entry->terminal;
+}
+
+gboolean
 desktop_entry_get_hidden (DesktopEntry *entry)
 {
   return (entry->flags & DESKTOP_ENTRY_HIDDEN) != 0;
diff -rupN gnome-menus-2.14.3-orig/libmenu/desktop-entries.h gnome-menus-2.14.3/libmenu/desktop-entries.h
--- gnome-menus-2.14.3-orig/libmenu/desktop-entries.h	2005-01-11 13:41:04.000000000 +0200
+++ gnome-menus-2.14.3/libmenu/desktop-entries.h	2021-10-01 18:50:21.716989350 +0300
@@ -48,6 +48,7 @@ const char *desktop_entry_get_name    (D
 const char *desktop_entry_get_comment (DesktopEntry *entry);
 const char *desktop_entry_get_icon    (DesktopEntry *entry);
 const char *desktop_entry_get_exec    (DesktopEntry *entry);
+gboolean    desktop_entry_get_launch_in_terminal (DesktopEntry *entry);
 
 gboolean desktop_entry_get_hidden         (DesktopEntry *entry);
 gboolean desktop_entry_get_no_display     (DesktopEntry *entry);
diff -rupN gnome-menus-2.14.3-orig/libmenu/gmenu-tree.c gnome-menus-2.14.3/libmenu/gmenu-tree.c
--- gnome-menus-2.14.3-orig/libmenu/gmenu-tree.c	2006-03-31 21:23:53.000000000 +0300
+++ gnome-menus-2.14.3/libmenu/gmenu-tree.c	2021-10-01 18:53:45.332781169 +0300
@@ -1090,6 +1090,14 @@ gmenu_tree_entry_get_exec (GMenuTreeEntr
   return desktop_entry_get_exec (entry->desktop_entry);
 }
 
+gboolean
+gmenu_tree_entry_get_launch_in_terminal (GMenuTreeEntry *entry)
+{
+  g_return_val_if_fail (entry != NULL, FALSE);
+
+  return desktop_entry_get_launch_in_terminal (entry->desktop_entry);
+}
+
 const char *
 gmenu_tree_entry_get_desktop_file_path (GMenuTreeEntry *entry)
 {
diff -rupN gnome-menus-2.14.3-orig/libmenu/gmenu-tree.h gnome-menus-2.14.3/libmenu/gmenu-tree.h
--- gnome-menus-2.14.3-orig/libmenu/gmenu-tree.h	2005-12-08 10:35:39.000000000 +0200
+++ gnome-menus-2.14.3/libmenu/gmenu-tree.h	2021-10-01 18:54:22.100774387 +0300
@@ -112,6 +112,7 @@ const char *gmenu_tree_entry_get_name
 const char *gmenu_tree_entry_get_comment (GMenuTreeEntry *entry);
 const char *gmenu_tree_entry_get_icon    (GMenuTreeEntry *entry);
 const char *gmenu_tree_entry_get_exec    (GMenuTreeEntry *entry);
+gboolean    gmenu_tree_entry_get_launch_in_terminal (GMenuTreeEntry *entry);
 
 const char *gmenu_tree_entry_get_desktop_file_path (GMenuTreeEntry *entry);
 const char *gmenu_tree_entry_get_desktop_file_id   (GMenuTreeEntry *entry);
