diff -rupN guess_fs-20140626-orig/guess_fstype.c guess_fs-20140626/guess_fstype.c
--- guess_fs-20140626-orig/guess_fstype.c	2021-03-19 17:10:20.201370030 +0200
+++ guess_fs-20140626/guess_fstype.c	2021-03-19 17:12:00.546370036 +0200
@@ -61,6 +61,8 @@ typedef unsigned int uint; /*BK*/
 #include <unistd.h>
 #include <sys/stat.h>
 #include <sys/types.h>
+#include <stdint.h>
+#include <linux/types.h>
 #include "linux_fs.h"
 #include "guess_fstype.h"
 //#include "nls.h"
