diff -rup checkinstall-1.6.2-orig/installwatch/installwatch.c checkinstall-1.6.2/installwatch/installwatch.c
--- checkinstall-1.6.2-orig/installwatch/installwatch.c	2020-12-25 20:35:53.264856675 +0200
+++ checkinstall-1.6.2/installwatch/installwatch.c	2020-12-25 20:39:47.027952613 +0200
@@ -3077,9 +3077,9 @@ int rmdir(const char *pathname) {
 	return result;
 }
 
-int scandir(	const char *dir,struct dirent ***namelist,
+int scandir(	const char *__restrict dir,struct dirent ***__restrict namelist,
 		int (*select)(const struct dirent *),
-		int (*compar)(const void *,const void *)	) {
+		int (*compar)(const struct dirent **,const struct dirent **)	) {
 	int result;
 
 	if (!libc_handle)
@@ -3689,9 +3689,9 @@ struct dirent64 *readdir64(DIR *dir) {
 	return result;
 }
 
-int scandir64(	const char *dir,struct dirent64 ***namelist,
+int scandir64(	const char *__restrict dir,struct dirent64 ***__restrict namelist,
 		int (*select)(const struct dirent64 *),
-		int (*compar)(const void *,const void *)	) {
+		int (*compar)(const struct dirent64 **, const struct dirent64 **)	) {
 	int result;
 
 	if (!libc_handle)
