diff -rup rox-filer-rox-filer-17w-orig/ROX-Filer/src/main.c rox-filer-rox-filer-17w/ROX-Filer/src/main.c
--- rox-filer-rox-filer-17w-orig/ROX-Filer/src/main.c	2020-12-26 10:16:16.334445704 +0200
+++ rox-filer-rox-filer-17w/ROX-Filer/src/main.c	2020-12-26 10:16:54.942134716 +0200
@@ -76,6 +76,8 @@
 #include "bulk_rename.h"
 #include "gtksavebox.h"
 
+extern gboolean session_auto_respawn;
+
 int number_of_windows = 0;	/* Quit when this reaches 0 again... */
 int to_wakeup_pipe = -1;	/* Write here to get noticed */
 
diff -rup rox-filer-rox-filer-17w-orig/ROX-Filer/src/session.h rox-filer-rox-filer-17w/ROX-Filer/src/session.h
--- rox-filer-rox-filer-17w-orig/ROX-Filer/src/session.h	2020-12-26 10:16:16.334445704 +0200
+++ rox-filer-rox-filer-17w/ROX-Filer/src/session.h	2020-12-26 10:16:39.554234288 +0200
@@ -10,8 +10,6 @@
 #include <gtk/gtk.h>
 #include <X11/SM/SMlib.h>
 
-gboolean session_auto_respawn;
-
 void session_init(const gchar *client_id);
 
 #endif
