--- linux-2.6.32.39/security/commoncap.c-orig	2011-05-10 17:57:30.476000018 +0200
+++ linux-2.6.32.39/security/commoncap.c	2011-05-10 17:58:56.463000017 +0200
@@ -593,6 +593,9 @@ int cap_bprm_secureexec(struct linux_bin
 		cred->egid != cred->gid);
 }
 
+/* needed by some third-party drivers, according to Barry Kauler */
+EXPORT_SYMBOL(cap_bprm_secureexec);
+
 /**
  * cap_inode_setxattr - Determine whether an xattr may be altered
  * @dentry: The inode/dentry being altered
