diff -rupN wlroots-0.13.0-orig/meson.build wlroots-0.13.0/meson.build
--- wlroots-0.13.0-orig/meson.build	2021-05-14 16:24:37.815318995 +0300
+++ wlroots-0.13.0/meson.build	2021-05-14 16:24:42.739340431 +0300
@@ -90,7 +90,7 @@ features = {
 	'xdg-foreign': false,
 }
 
-wayland_server = dependency('wayland-server', version: '>=1.19')
+wayland_server = dependency('wayland-server', version: '>=1.18')
 wayland_client = dependency('wayland-client')
 wayland_protos = dependency('wayland-protocols', version: '>=1.17')
 egl = dependency('egl')
