diff -rupN wlroots-0.13.0-orig/meson.build wlroots-0.13.0/meson.build
--- wlroots-0.13.0-orig/meson.build	2021-05-14 16:24:37.815318995 +0300
+++ wlroots-0.13.0/meson.build	2021-05-14 16:24:42.739340431 +0300
@@ -90,7 +90,7 @@ features = {
 	'xdg-foreign': false,
 }
 
-wayland_server = dependency('wayland-server', version: '>=1.19')
+wayland_server = dependency('wayland-server', version: '>=1.18')
 wayland_client = dependency('wayland-client')
 wayland_protos = dependency('wayland-protocols', version: '>=1.17')
 egl = dependency('egl')
diff -rupN wlroots-0.13.0-orig/types/seat/wlr_seat.c wlroots-0.13.0/types/seat/wlr_seat.c
--- wlroots-0.13.0-orig/types/seat/wlr_seat.c	2021-05-14 16:24:37.815318995 +0300
+++ wlroots-0.13.0/types/seat/wlr_seat.c	2021-05-14 17:41:25.413232742 +0300
@@ -20,7 +20,7 @@ static void seat_handle_get_pointer(stru
 	struct wlr_seat_client *seat_client =
 		wlr_seat_client_from_resource(seat_resource);
 	if (!(seat_client->seat->accumulated_capabilities & WL_SEAT_CAPABILITY_POINTER)) {
-		wl_resource_post_error(seat_resource, WL_SEAT_ERROR_MISSING_CAPABILITY,
+		wl_resource_post_error(seat_resource, 0,
 				"wl_seat.get_pointer called when no pointer capability has existed");
 		return;
 	}
@@ -34,7 +34,7 @@ static void seat_handle_get_keyboard(str
 	struct wlr_seat_client *seat_client =
 		wlr_seat_client_from_resource(seat_resource);
 	if (!(seat_client->seat->accumulated_capabilities & WL_SEAT_CAPABILITY_KEYBOARD)) {
-		wl_resource_post_error(seat_resource, WL_SEAT_ERROR_MISSING_CAPABILITY,
+		wl_resource_post_error(seat_resource, 0,
 				"wl_seat.get_keyboard called when no keyboard capability has existed");
 		return;
 	}
@@ -48,7 +48,7 @@ static void seat_handle_get_touch(struct
 	struct wlr_seat_client *seat_client =
 		wlr_seat_client_from_resource(seat_resource);
 	if (!(seat_client->seat->accumulated_capabilities & WL_SEAT_CAPABILITY_TOUCH)) {
-		wl_resource_post_error(seat_resource, WL_SEAT_ERROR_MISSING_CAPABILITY,
+		wl_resource_post_error(seat_resource, 0,
 				"wl_seat.get_touch called when no touch capability has existed");
 		return;
 	}
