diff -rupN wlroots-0.14.0-orig/meson.build wlroots-0.14.0/meson.build
--- wlroots-0.14.0-orig/meson.build	2021-06-23 21:11:00.576321097 +0300
+++ wlroots-0.14.0/meson.build	2021-06-23 21:11:22.872314355 +0300
@@ -96,7 +96,7 @@ internal_features = {
 
 wayland_server = dependency('wayland-server', version: '>=1.19')
 wayland_client = dependency('wayland-client')
-drm = dependency('libdrm', version: '>=2.4.105')
+drm = dependency('libdrm', version: '>=2.4.104')
 gbm = dependency('gbm', version: '>=17.1.0')
 libinput = dependency('libinput', version: '>=1.14.0')
 xkbcommon = dependency('xkbcommon')
