From 059727dbb98282eb76eb352212caac8c1549ea03 Mon Sep 17 00:00:00 2001
From: Dima Krasner <dima@dimakrasner.com>
Date: Sun, 16 May 2021 19:20:16 +0300
Subject: [PATCH] meson: link with libinput

---
 meson.build | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/meson.build b/meson.build
index a268f1a..0669e32 100644
--- a/meson.build
+++ b/meson.build
@@ -39,6 +39,7 @@ wayland_protos = dependency('wayland-protocols', version: '>=1.14')
 wayland_server = dependency('wayland-server')
 pixman         = dependency('pixman-1')
 xkbcommon      = dependency('xkbcommon')
+libinput       = dependency('libinput')
 math           = cc.find_library('m')
 
 wl_protocol_dir = wayland_protos.get_pkgconfig_variable('pkgdatadir')
@@ -158,6 +159,7 @@ executable(
     wlroots,
     xkbcommon,
     pixman,
+    libinput,
     math,
   ],
   install: true,
-- 
2.25.1

