diff -x .git -rupN cage-orig/output.c cage/output.c
--- cage-orig/output.c	2021-09-28 09:59:32.039680544 +0000
+++ cage/output.c	2021-09-28 09:59:37.556235981 +0000
@@ -464,10 +464,26 @@ handle_new_output(struct wl_listener *li
 	output->damage_destroy.notify = handle_output_damage_destroy;
 	wl_signal_add(&output->damage->events.destroy, &output->damage_destroy);
 
-	struct wlr_output_mode *preferred_mode = wlr_output_preferred_mode(wlr_output);
-	if (preferred_mode) {
-		wlr_output_set_mode(wlr_output, preferred_mode);
+	if (!wl_list_empty(&wlr_output->modes)) {
+		struct wlr_output_mode *preferred_mode = wlr_output_preferred_mode(wlr_output);
+		if (preferred_mode) {
+			wlr_output_set_mode(wlr_output, preferred_mode);
+		}
+		if (!wlr_output_test(wlr_output)) {
+			struct wlr_output_mode *mode;
+			wl_list_for_each(mode, &wlr_output->modes, link) {
+				if (mode == preferred_mode) {
+					continue;
+				}
+
+				wlr_output_set_mode(wlr_output, mode);
+				if (wlr_output_test(wlr_output)) {
+					break;
+				}
+			}
+		}
 	}
+
 	wlr_output_set_transform(wlr_output, output->server->output_transform);
 
 	if (server->output_mode == CAGE_MULTI_OUTPUT_MODE_LAST) {
