diff -rupN --exclude .git cage-clean/output.c cage/output.c
--- cage-clean/output.c	2021-09-30 19:51:28.904027424 +0800
+++ cage/output.c	2021-09-30 19:51:02.636026799 +0800
@@ -246,7 +246,19 @@ output_enable(struct cg_output *output)
 	 * duplicate the enabled property in cg_output. */
 	wlr_log(WLR_DEBUG, "Enabling output %s", wlr_output->name);
 
-	wlr_output_layout_add_auto(output->server->output_layout, wlr_output);
+	/* Add the new output to the right of the rightmost display */
+	int max_x = 0, max_x_y = 0;
+	struct wlr_output_layout_output *l_output;
+	wl_list_for_each(l_output, &output->server->output_layout->outputs, link) {
+		int width, height;
+		wlr_output_effective_resolution(l_output->output, &width, &height);
+		if (l_output->x + width > max_x) {
+			max_x = l_output->x + width;
+			max_x_y = l_output->y;
+		}
+	}
+
+	wlr_output_layout_add(output->server->output_layout, wlr_output, max_x, max_x_y);
 	wlr_output_enable(wlr_output, true);
 	wlr_output_commit(wlr_output);
 }
