diff -rupN cage-0.1.3-orig/cage.c cage-0.1.3/cage.c
--- cage-0.1.3-orig/cage.c	2021-04-16 15:01:38.000000000 +0300
+++ cage-0.1.3/cage.c	2021-05-14 18:42:51.883294958 +0300
@@ -145,6 +145,8 @@ cleanup_primary_client(pid_t pid)
 static bool
 drop_permissions(void)
 {
+	return true;
+
 	if (getuid() != geteuid() || getgid() != getegid()) {
 		// Set the gid and uid in the correct order.
 		if (setgid(getgid()) != 0 || setuid(getuid()) != 0) {
