diff -rupN libxcvt-0.1.1-orig/lib/meson.build libxcvt-0.1.1/lib/meson.build
--- libxcvt-0.1.1-orig/lib/meson.build	2022-01-20 09:54:00.551992942 +0200
+++ libxcvt-0.1.1/lib/meson.build	2022-01-20 09:54:16.627992864 +0200
@@ -1,6 +1,6 @@
 libxcvt_sources = ['libxcvt.c']
-libxcvt = shared_library('xcvt',
-                         libxcvt_sources,
-                         include_directories : inc,
-                         version: meson.project_version(),
-                         install : true)
+libxcvt = library('xcvt',
+                  libxcvt_sources,
+                  include_directories : inc,
+                  version: meson.project_version(),
+                  install : true)
