diff -rup cage-orig/view.c cage/view.c
--- cage-orig/view.c	2021-06-23 20:53:14.906968344 +0300
+++ cage/view.c	2021-06-23 20:53:57.865787400 +0300
@@ -235,7 +235,10 @@ view_map(struct cg_view *view, struct wl
 	view->wlr_surface = surface;
 
 	struct wlr_subsurface *subsurface;
-	wl_list_for_each (subsurface, &view->wlr_surface->subsurfaces, parent_link) {
+	wl_list_for_each (subsurface, &view->wlr_surface->subsurfaces_below, parent_link) {
+		subsurface_create(view, subsurface);
+	}
+	wl_list_for_each (subsurface, &view->wlr_surface->subsurfaces_above, parent_link) {
 		subsurface_create(view, subsurface);
 	}
 
