diff -rup mtPaint-199472ad6a4ecee6c8583fb5a504a2e99712b4fc-orig/configure mtPaint-199472ad6a4ecee6c8583fb5a504a2e99712b4fc/configure
--- mtPaint-199472ad6a4ecee6c8583fb5a504a2e99712b4fc-orig/configure	2024-08-12 20:03:32.156855211 +0300
+++ mtPaint-199472ad6a4ecee6c8583fb5a504a2e99712b4fc/configure	2024-08-12 20:03:50.516866792 +0300
@@ -619,7 +619,7 @@ then
 		LIB_NAME "" "$STATIC_FT" "`PKGCONFIG freetype2 --libs`"
 	fi
 	# libiconv may be separate, or part of libc
-	HAVE_FUNC "iconv_open" || LIB_NAME iconv "$STATIC_FT"
+	HAVE_FUNC "iconv_open"
 fi
 if [ "$NCMS" = LittleCMS2 ]
 then
@@ -642,7 +642,7 @@ LIB_NAME z "$STATIC_PNG"
 LIB_NAME m
 if [ "$MT_LANG" = YES ]
 then
-	HAVE_FUNC "gettext" || LIB_NAME intl
+	HAVE_FUNC "gettext"
 	DEFS="$DEFS -DU_NLS -DMT_LANG_DEST='\"$MT_PO\"'"
 fi
 
