diff -rupN freememapplet-2.8.6-org/freememapplet_tray.c freememapplet-2.8.6/freememapplet_tray.c
--- freememapplet-2.8.6-org/freememapplet_tray.c	2021-09-25 11:12:28.819925060 +0300
+++ freememapplet-2.8.6/freememapplet_tray.c	2021-09-25 11:12:37.351907749 +0300
@@ -162,7 +162,7 @@ void tray_icon_on_menu(GtkStatusIcon *st
 	gtk_menu_shell_append(GTK_MENU_SHELL(menu), menuitem);
 	
 	gtk_widget_show_all(menu);
-	gtk_menu_popup(GTK_MENU(menu), NULL, NULL, NULL, NULL, button, gdk_event_get_time(NULL));
+	gtk_menu_popup(GTK_MENU(menu), NULL, NULL, NULL, NULL, button, gtk_get_current_event_time());
 }
 
 static GtkStatusIcon *create_tray_icon() {
