diff -rup freememapplet-2.8.6-orig/freememapplet_tray.c freememapplet-2.8.6/freememapplet_tray.c
--- freememapplet-2.8.6-orig/freememapplet_tray.c	2021-01-20 19:28:00.592854207 +0200
+++ freememapplet-2.8.6/freememapplet_tray.c	2021-01-20 19:28:35.672654453 +0200
@@ -97,17 +97,15 @@ gboolean Update(gpointer ptr) {
 		g_snprintf(memdisplaylong, MAXCHARMSG, "%s %s %s\n%s", memdisplaytotal, STORAGE_MSG, memdisplayfree, RIGHT_MENU);
     else
 		g_snprintf(memdisplaylong, MAXCHARMSG, "%s %s %s", memdisplaytotal, STORAGE_MSG, memdisplayfree);
-    gtk_status_icon_set_tooltip(tray_icon, memdisplaylong);
+    gtk_status_icon_set_tooltip_text(tray_icon, memdisplaylong);
 
     //update icon... (sizefree,sizetotal are in MB)
-    if (gtk_status_icon_get_blinking(tray_icon)==TRUE) gtk_status_icon_set_blinking(tray_icon,FALSE);
 	if (sizetotal == 0)
 		percentfree = 0;
 	else
 		percentfree=(sizefree*100)/sizetotal;
     if (sizefree < 20) {
         gtk_status_icon_set_from_pixbuf(tray_icon,critical_pixbuf);
-        gtk_status_icon_set_blinking(tray_icon,TRUE);
     }
     else if (sizefree < 50) {
         gtk_status_icon_set_from_pixbuf(tray_icon,critical_pixbuf);
