diff -rupN gmeasures-0.7-orig/src/callbacks.c gmeasures-0.7/src/callbacks.c
--- gmeasures-0.7-orig/src/callbacks.c	2024-08-12 20:38:44.681138729 +0300
+++ gmeasures-0.7/src/callbacks.c	2024-08-12 20:38:57.001145224 +0300
@@ -287,7 +287,7 @@ void on_editdlg_entry_unit2_changed (Gtk
 
 void on_sb_entry_cdegrees_changed (GtkEditable *editable, gpointer user_data)
 {
-	if(gtk_widget_has_focus(editable)) {
+	if(gtk_widget_has_focus(GTK_WIDGET(editable))) {
 		Mode = 1;
 		AppCalculate();
 	}
@@ -300,7 +300,7 @@ void on_sb_entry_cdegrees_grab_focus (Gt
 
 void on_sb_entry_fdegrees_changed (GtkEditable *editable, gpointer user_data)
 {
-	if(gtk_widget_has_focus(editable)) {
+	if(gtk_widget_has_focus(GTK_WIDGET(editable))) {
 		Mode = 2;
 		AppCalculate();
 	}
diff -rupN gmeasures-0.7-orig/src/interface.c gmeasures-0.7/src/interface.c
--- gmeasures-0.7-orig/src/interface.c	2024-08-12 20:38:44.681138729 +0300
+++ gmeasures-0.7/src/interface.c	2024-08-12 20:38:57.001145224 +0300
@@ -41,7 +41,7 @@ create_App (void)
   GtkWidget *label4;
   GtkWidget *label28;
   GtkWidget *entry_val2;
-  GtkWidget *sb_entry_val1_adj;
+  GtkAdjustment *sb_entry_val1_adj;
   GtkWidget *sb_entry_val1;
   GtkWidget *combo_unit1;
   GtkWidget *combo_unit2;
@@ -55,9 +55,9 @@ create_App (void)
   GtkWidget *hbox12;
   GtkWidget *label11;
   GtkWidget *label24;
-  GtkWidget *sb_entry_cdegrees_adj;
+  GtkAdjustment *sb_entry_cdegrees_adj;
   GtkWidget *sb_entry_cdegrees;
-  GtkWidget *sb_entry_fdegrees_adj;
+  GtkAdjustment *sb_entry_fdegrees_adj;
   GtkWidget *sb_entry_fdegrees;
   GtkWidget *label8;
   GtkWidget *label20;
diff -rupN gmeasures-0.7-orig/src/main.c gmeasures-0.7/src/main.c
--- gmeasures-0.7-orig/src/main.c	2024-08-12 20:38:44.681138729 +0300
+++ gmeasures-0.7/src/main.c	2024-08-12 20:38:57.001145224 +0300
@@ -340,8 +340,8 @@ void AppUpdateUnit2 ()
 	InSetup = 1;
 	combo_unit1 = (GtkComboBoxText *) lookup_widget(GTK_WIDGET(App), "combo_unit1");
 	combo_unit2 = (GtkComboBoxText *) lookup_widget(GTK_WIDGET(App), "combo_unit2");
-	unit1 = ComboGetActiveText(combo_unit1);
-	model = gtk_combo_box_get_model(combo_unit2);
+	unit1 = ComboGetActiveText(GTK_COMBO_BOX(combo_unit1));
+	model = gtk_combo_box_get_model(GTK_COMBO_BOX(combo_unit2));
 	cnt = gtk_tree_model_iter_n_children(model, NULL);
 	for(i = cnt; i >= 0; i--)  gtk_combo_box_text_remove(combo_unit2, i);
 	for(i = 0; i < Msr->len; i++) {
@@ -352,7 +352,7 @@ void AppUpdateUnit2 ()
 	}
 	g_free(unit1);
 	gtk_tree_sortable_set_sort_column_id(GTK_TREE_SORTABLE(model), LIST_UNIT1, GTK_SORT_ASCENDING);
-	gtk_combo_box_set_active(combo_unit2, 0);
+	gtk_combo_box_set_active(GTK_COMBO_BOX(combo_unit2), 0);
 	AppCalculate();
 	InSetup = 0;
 }
