--- aufs3-standalone.patch	2017-03-26 17:23:50.450649114 +0800
+++ aufs3-standalone.patch-patched	2017-03-26 17:57:57.807413874 +0800
@@ -1,18 +1,5 @@
 aufs3.4 standalone patch
 
-diff --git a/fs/file_table.c b/fs/file_table.c
-index 70f2a0f..146a3d7 100644
---- a/fs/file_table.c
-+++ b/fs/file_table.c
-@@ -442,6 +442,8 @@ void file_sb_list_del(struct file *file)
- 	}
- }
- 
-+EXPORT_SYMBOL(file_sb_list_del);
-+
- #ifdef CONFIG_SMP
- 
- /*
 diff --git a/fs/inode.c b/fs/inode.c
 index 9f4f5fe..bb0f3ba 100644
 --- a/fs/inode.c
