diff -Naur aufs3.4/aufs3-standalone.patch aufs3.4-patched/aufs3-standalone.patch
--- aufs3.4/aufs3-standalone.patch	2017-04-16 12:01:59.290528000 +0800
+++ aufs3.4-patched/aufs3-standalone.patch	2017-04-16 12:13:32.440562160 +0800
@@ -1,17 +1,5 @@
 aufs3.4 standalone patch
 
-diff --git a/fs/file_table.c b/fs/file_table.c
-index 70f2a0f..146a3d7 100644
---- a/fs/file_table.c
-+++ b/fs/file_table.c
-@@ -442,6 +442,8 @@ void file_sb_list_del(struct file *file)
- 	}
- }
- 
-+EXPORT_SYMBOL(file_sb_list_del);
-+
- #ifdef CONFIG_SMP
- 
  /*
 diff --git a/fs/inode.c b/fs/inode.c
 index 9f4f5fe..bb0f3ba 100644
diff -Naur aufs3.4/fs/aufs/dir.c aufs3.4-patched/fs/aufs/dir.c
--- aufs3.4/fs/aufs/dir.c	2017-04-16 12:01:59.350528000 +0800
+++ aufs3.4-patched/fs/aufs/dir.c	2017-04-16 12:14:11.513897383 +0800
@@ -225,9 +225,6 @@
 	finfo = au_fi(file);
 	fidir = finfo->fi_hdir;
 	if (fidir) {
-		/* remove me from sb->s_files */
-		file_sb_list_del(file);
-
 		vdir_cache = fidir->fd_vdir_cache; /* lock-free */
 		if (vdir_cache)
 			au_vdir_free(vdir_cache);
diff -Naur aufs3.4/fs/aufs/f_op.c aufs3.4-patched/fs/aufs/f_op.c
--- aufs3.4/fs/aufs/f_op.c	2017-04-16 12:01:59.357195000 +0800
+++ aufs3.4-patched/fs/aufs/f_op.c	2017-04-16 12:14:38.383898682 +0800
@@ -84,8 +84,6 @@
 	finfo = au_fi(file);
 	bindex = finfo->fi_btop;
 	if (bindex >= 0) {
-		/* remove me from sb->s_files */
-		file_sb_list_del(file);
 		au_set_h_fptr(file, bindex, NULL);
 	}
 
diff -Naur aufs3.4/fs/aufs/vfsub.h aufs3.4-patched/fs/aufs/vfsub.h
--- aufs3.4/fs/aufs/vfsub.h	2017-04-16 12:01:59.460528000 +0800
+++ aufs3.4-patched/fs/aufs/vfsub.h	2017-04-16 12:13:59.010563445 +0800
@@ -32,7 +32,6 @@
 /* copied from linux/fs/internal.h */
 /* todo: BAD approach!! */
 DECLARE_BRLOCK(vfsmount_lock);
-extern void file_sb_list_del(struct file *f);
 extern spinlock_t inode_sb_list_lock;
 
 /* copied from linux/fs/file_table.c */
