diff --git a/fs/aufs/i_op.c b/fs/aufs/i_op.c
index c3ff23f485d7..329d73968dc3 100644
--- a/fs/aufs/i_op.c
+++ b/fs/aufs/i_op.c
@@ -1293,9 +1293,13 @@ 
		goto out_fill; /* pretending success */

	positive = d_is_positive(h_path.dentry);
-	if (positive)
+	if (positive) {
		/* no vfsub version */
-		err = vfs_getattr(&h_path, st, request, query);
+ 		if (query & AT_GETATTR_NOSEC)
+ 		err = vfs_getattr_nosec(&h_path, st, request, query);
+ 		else
+ 		err = vfs_getattr(&h_path, st, request, query);
+ 	}
	if (!err) {
		if (positive)
			au_refresh_iattr(inode, st,
