diff -rupN yambar-orig/modules/meson.build yambar/modules/meson.build
--- yambar-orig/modules/meson.build	2021-11-17 09:33:01.279875813 +0200
+++ yambar/modules/meson.build	2021-11-17 09:33:04.771875700 +0200
@@ -5,7 +5,7 @@ modules = []
 alsa = dependency('alsa')
 udev = dependency('libudev')
 json = dependency('json-c')
-mpd = dependency('libmpdclient')
+mpd = dependency('libmpdclient', required: false)
 xcb_xkb = dependency('xcb-xkb', required: get_option('backend-x11'))
 
 # Module name -> (source-list, dep-list)
@@ -16,7 +16,6 @@ mod_data = {
   'clock': [[], []],
   'i3': [['i3-common.c', 'i3-common.h'], [dynlist, json]],
   'label': [[], []],
-  'mpd': [[], [mpd]],
   'network': [[], []],
   'removables': [[], [dynlist, udev]],
   'script': [[], []],
@@ -30,6 +29,12 @@ if backend_x11
   }
 endif
 
+if mpd.found()
+  mod_data += {
+    'mpd': [[], [mpd]],
+  }
+endif
+
 if backend_wayland
   river_proto_headers = []
   river_proto_src = []
diff -rupN yambar-orig/plugin.c yambar/plugin.c
--- yambar-orig/plugin.c	2021-11-17 09:33:01.279875813 +0200
+++ yambar/plugin.c	2021-11-17 09:33:59.343873939 +0200
@@ -39,7 +39,9 @@ EXTERN_MODULE(clock);
 EXTERN_MODULE(foreign_toplevel);
 EXTERN_MODULE(i3);
 EXTERN_MODULE(label);
+#if defined(HAVE_PLUGIN_mpd)
 EXTERN_MODULE(mpd);
+#endif
 EXTERN_MODULE(network);
 EXTERN_MODULE(removables);
 EXTERN_MODULE(river);
@@ -117,7 +119,9 @@ init(void)
 #endif
     REGISTER_CORE_MODULE(i3, i3);
     REGISTER_CORE_MODULE(label, label);
+#if defined(HAVE_PLUGIN_mpd)
     REGISTER_CORE_MODULE(mpd, mpd);
+#endif
     REGISTER_CORE_MODULE(network, network);
     REGISTER_CORE_MODULE(removables, removables);
 #if defined(HAVE_PLUGIN_river)
