diff -rupN yambar-orig/subprojects/fcft.wrap yambar/subprojects/fcft.wrap
--- yambar-orig/subprojects/fcft.wrap	2022-02-09 18:14:06.272236012 +0000
+++ yambar/subprojects/fcft.wrap	2022-02-09 18:14:12.744691592 +0000
@@ -1,3 +1,3 @@
 [wrap-git]
 url = https://codeberg.org/dnkl/fcft.git
-revision = master
+revision = 7a862f66d38973357399f9eef4e8d9342d7bc5a4
diff -rupN yambar-orig/subprojects/tllist.wrap yambar/subprojects/tllist.wrap
--- yambar-orig/subprojects/tllist.wrap	2022-02-09 18:14:06.272236012 +0000
+++ yambar/subprojects/tllist.wrap	2022-02-09 18:14:36.726379602 +0000
@@ -1,3 +1,3 @@
 [wrap-git]
 url = https://codeberg.org/dnkl/tllist.git
-revision = master
+revision = e6670483da553e85ebfe24ae689c6f68e207ef04
