diff -rup transmission-3.00-orig/gtk/conf.c transmission-3.00/gtk/conf.c
--- transmission-3.00-orig/gtk/conf.c	2022-01-07 21:24:00.198966531 +0800
+++ transmission-3.00/gtk/conf.c	2022-01-07 21:24:54.034964794 +0800
@@ -66,11 +66,6 @@ static void tr_prefs_init_defaults(tr_va
 
     if (dir == NULL)
     {
-        dir = g_get_user_special_dir(G_USER_DIRECTORY_DESKTOP);
-    }
-
-    if (dir == NULL)
-    {
         dir = tr_getDefaultDownloadDir();
     }
 
diff -rup transmission-3.00-orig/gtk/makemeta-ui.c transmission-3.00/gtk/makemeta-ui.c
--- transmission-3.00-orig/gtk/makemeta-ui.c	2022-01-07 21:24:00.198966531 +0800
+++ transmission-3.00/gtk/makemeta-ui.c	2022-01-07 21:24:28.774965609 +0800
@@ -378,7 +378,7 @@ static void onFileToggled(GtkToggleButto
 
 static char const* getDefaultSavePath(void)
 {
-    return g_get_user_special_dir(G_USER_DIRECTORY_DESKTOP);
+    return g_get_user_special_dir(G_USER_DIRECTORY_DOWNLOAD);
 }
 
 static void on_drag_data_received(GtkWidget* widget UNUSED, GdkDragContext* drag_context, gint x UNUSED, gint y UNUSED,
