diff -rupN transmission-4.0.0-orig/gtk/MakeDialog.cc transmission-4.0.0/gtk/MakeDialog.cc
--- transmission-4.0.0-orig/gtk/MakeDialog.cc	2023-02-08 19:48:54.372272138 +0200
+++ transmission-4.0.0/gtk/MakeDialog.cc	2023-02-08 19:50:02.730558700 +0200
@@ -533,7 +533,7 @@ MakeDialog::Impl::Impl(MakeDialog& dialo
 {
     dialog_.signal_response().connect(sigc::mem_fun(*this, &Impl::onResponse));
 
-    destination_chooser_->set_filename(Glib::get_user_special_dir(TR_GLIB_USER_DIRECTORY(DESKTOP)));
+    destination_chooser_->set_filename(Glib::get_user_special_dir(TR_GLIB_USER_DIRECTORY(DOWNLOAD)));
 
     folder_radio_->signal_toggled().connect([this]() { onSourceToggled(folder_radio_, folder_chooser_); });
     folder_chooser_->signal_selection_changed().connect([this]() { onChooserChosen(folder_chooser_); });
diff -rupN transmission-4.0.0-orig/gtk/Prefs.cc transmission-4.0.0/gtk/Prefs.cc
--- transmission-4.0.0-orig/gtk/Prefs.cc	2023-02-08 19:48:54.372272138 +0200
+++ transmission-4.0.0/gtk/Prefs.cc	2023-02-08 19:49:49.906835626 +0200
@@ -41,11 +41,6 @@ static void tr_prefs_init_defaults(tr_va
 
     if (dir.empty())
     {
-        dir = Glib::get_user_special_dir(TR_GLIB_USER_DIRECTORY(DESKTOP));
-    }
-
-    if (dir.empty())
-    {
         dir = tr_getDefaultDownloadDir();
     }
 
